---
title: "A Usage Template for the R Package msDiaLogue"
output: rmarkdown::html_vignette
date: "`r Sys.Date()`"
bibliography:
  - ../inst/bib/msDiaLogue.bib
vignette: >
  %\VignetteIndexEntry{A Usage Template}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>", fig.align = "center")
```

## Load R package

```{r warning=FALSE, message=FALSE}
library(msDiaLogue)
```


## Preprocessing

The function `preprocessing()` takes a `.csv` file of summarized protein abundances,
exported from **Spectronaut**. The most important columns that need to be included in this
file are: `R.Condition`, `R.Replicate`, `PG.ProteinAccessions`, `PG.ProteinNames`,
`PG.IsSingleHit`, and `PG.Quantity`. This function will reformat the data and provide
functionality for some initial filtering (based on the number of unique peptides).

**1\.** Load the raw data

+ If the raw data is in a .csv file
[Toy_Spectronaut_Data.csv](https://github.com/uconn-scs/msDiaLogue/blob/main/tests/testData/Toy_Spectronaut_Data.csv),
specify the `fileName` to read the raw data file into **R**.

+ If the raw data is stored as an .RData file
[Toy_Spectronaut_Data.RData](https://github.com/uconn-scs/msDiaLogue/blob/main/tests/testData/Toy_Spectronaut_Data.RData),
first load the data file directly, then specify the `dataSet` in the function.
     
**2\.** Whether to omit observations with NaN

NaN, which stands for 'Not a Number,' represents to protein entries in each condition that
lack a measured quantitative value.

**3\.** The number of unique peptides are required to include a protein

The default is to filter out proteins with fewer than 2 unique peptides. Removing proteins
identified by only 1 unique peptide sequence is more likely to be representatives of the
1% false positive protein identifications. We choose to remove these protein entries from
our downstream analysis in order to provide more confident quantitative results across the
identified proteome. However, these can still be observed in the original protein report
from **Spectronaut**.

**4\.** Whether to rename proteins without names by their accession numbers

**5\.** Whether to save removed data to current working directory

If `saveRm = TRUE`, the data removed in step 2 (*preprocess_Filtered_Out_NaN.csv*) and
step 3 (*preprocess_Filtered_Out_Unique.csv*) will be saved in the current working
directory.

As part of the `preprocessing()`, a histogram of $log_2$-transformed protein abundances is
provided. This is a helpful way to confirm that the data have been read in correctly, and
there are no issues with the numerical values of the protein abundances. Ideally, this
histogram will appear fairly symmetrical (bell-shaped) without too much skew towards
smaller or larger values.

```{r eval=FALSE}
## if the raw data is in a .csv file
fileName <- "../tests/testData/Toy_Spectronaut_Data.csv"
dataSet <- preprocessing(fileName,
                         filterNaN = TRUE, filterUnique = 2,
                         replaceBlank = TRUE, saveRm = TRUE)
```

**NOTE:** `preprocessing()` does not perform a transformation on your data. You still need
to use the functon `transform()`.

```{r results='hide'}
## if the raw data is in an .Rdata file
load("../tests/testData/Toy_Spectronaut_Data.RData")
dataSet <- preprocessing(dataSet = Toy_Spectronaut_Data,
                         filterNaN = TRUE, filterUnique = 2,
                         replaceBlank = TRUE, saveRm = TRUE)
```

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(dataSet, align = "l")
```
</div>


## Transformation

Raw intensity measurements, often unsuitable for direct statistical modeling, prompt
transformation in most quantitative proteomics workflow. The preferred solution is
log-transformation, effectively addressing this issue. The log$_2$ transformation is
commonly employed due to its ability to facilitate the straightforward interpretation of
fold changes in protein levels.

```{r}
dataTran <- transform(dataSet, logFold = 2)
```

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(dataTran)
```
</div>


## Filtering

For various reasons, such as the presence of high-abundance contaminant proteins that may
be irrelevant to the present analysis, a lack of informativeness (e.g., proteins that
exhibit no differential abundance at all or proteins that are known with certainty to be
false positive identifications with more than one unique peptide identified),
identification of a protein exclusively in a small subset of samples or only one
experimental condition, etc., we recommend data filtering before conducting differential
analysis. The application of protein-level filters is optional and is expected to vary by
project.

**Case 1.** Remove the listed proteins and retain the remaining proteins

For example, the protein "XPO4_HUMAN" is chosen to be filtered out.

```{r eval=FALSE}
filterOutIn(dataTran, listName = "XPO4_HUMAN",
            removeList = TRUE, saveRm = TRUE)
```

where `removeList = TRUE` indicates the removal of proteins listed in `listName` from the
`dataTran`. Please note that if `saveRm = TRUE`, the excluded data ("XPO4_HUMAN") will be
saved as a .csv file named *filtered_out_data.csv* in the current working directory.

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(
  filterOutIn(dataTran, listName = "XPO4_HUMAN", removeList = TRUE, saveRm = TRUE))
```
</div>

**Case 2.** Keep the listed proteins and remove the rest.

Alternatively, if we would to keep proteins like "PA1B2_HUMAN" and "TEBP_HUMAN", simply
set `removelist = FALSE`.

```{r eval=FALSE}
filterOutIn(dataTran, listName = c("PA1B2_HUMAN", "TEBP_HUMAN"),
            removeList = FALSE)
```

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(filterOutIn(
  dataTran, listName = c("PA1B2_HUMAN", "TEBP_HUMAN"), removeList = FALSE))
```
</div>


## Normalization

Normalization is designed to address systematic biases, enhancing the comparability of
samples while preserving the signal. Various normalization approaches have been proposed.
So far, this package provides three normalization methods for use:

1. "quant": Quantile [@bolstad2003comparison]

2. "median": Protein-wise Median

3. "mean": Protein-wise Mean

Quantile normalization is generally recommended.

```{r}
dataNorm <- normalize(dataTran, normalizeType = "quant")
```

Oh! The message "Warning: Removed 16 rows containing non-finite values" indicates the
presence of 16 NA (Not Available) values in the data. These NA values are are
automatically excluded when generating the boxplot but retained in the actual dataset.

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(dataNorm)
```
</div>


## Imputation

Traditional Data-Dependent Acquisition (DDA) mass spectrometry-based quantitative
proteomics datasets often yield many missing quantitative values due to limits in sampling
frequency. Data-Independent Acquisition (DIA) helps to substantially reduce the number of
missing values compared to DDA experiments, but some will still exist due to sample
variability, and these missing values will impact downstream analyses.

Function `dataMissing()` is designed to summarize the missingness for each protein, where
`plot = TRUE` indicates plotting the missingness, and `show_labels = TRUE` means that the
protein names are displayed in the printed plot. Note that the visual representation is
not generated by default, and the plot generation time varies with project size.

```{r}
dataMissing <- dataMissing(dataNorm, plot = TRUE, show_labels = TRUE)
```

The percentage in the protein labels represents the proportion of missing data in the
samples for that protein. For instance, the label "XPO4_HUMAN (80\%)" indicates that,
within all observations for the protein "XPO4_HUMAN", 80\% of the data is missing.
Additionally, the percentage in the legend represents the proportion of missing data in
the whole dataset. In this case, 8.4\% of the data in `dataNorm` is missing.

Regardless of plot generation, the function `dataMissing()` always returns a table
providing the following information:

+ `count_miss`: The count of missing values for each protein.

+ `pct-miss`: The percentage of missing values for each protein.

+ `pct_total_miss`: The percentage of missing values for each protein relative to the
total missing values in the entire dataset.

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(dataMissing)
```
</div>

For example, in the case of the protein "XPO4_HUMAN," there are 8 NA values in the samples, representing 80\% of the missing data for "XPO4_HUMAN" within that sample and 50\% of the
total missing data in the entire dataset.

Various imputation methods have been developed to address the missing-value issue and
assign a quantitative value to proteins with missing quantitative values. So far, this
package provides five imputation methods for use:

1. "LocalMinVal": Replaces missing values with the lowest value from the corresponding
protein by condition combination;

2. "GlobalMinVal": Replaces missing values with the lowest value found within the entire
dataset.

3. "knn": Replaces missing values using the k-nearest neighbors algorithm
[@troyanskaya2001missing].

4. "seq-knn": Replaces missing values using the sequential k-nearest neighbors algorithm
[@kim2004reuse].

5. "trunc-knn": Replaces missing values using the truncated k-nearest neighbors algorithm
[@shah2017distribution].

Additional methods will be added later. And "LocalMinVal" is currently the default method.

For example, to impute the NA value of `dataNorm` using "LocalMinVal," set the required
percentage of values that must be present in a given protein by condition combination for
values to be imputed to 51\%.

```{r}
dataImput <- impute(dataNorm, imputeType = "LocalMinVal",
                    reqPercentPresent = 0.51, reportImputing = FALSE)
```

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(dataImput)
```
</div>

If `reportImputing = TRUE`, the returned result structure will be altered to a list,
adding a shadow data frame with imputed data labels, where 1 indicates the corresponding
entries have been imputed, and 0 indicates otherwise.

After the above imputation, there may still be some NA values that consequently need to be
filtered out.

```{r}
dataImput <- filterNA(dataImput, saveRm = TRUE)
```

where `saveRm = TRUE` indicates that the filtered data will be saved as a .csv file named
*filtered_NA_data.csv* in the current working directory.

The `dataImput` is as follows:

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(dataImput)
```
</div>


## Summarization

Summarization proves valuable in distilling meaningful insights from vast and intricate
data, facilitating the efficient identification, quantification, and interpretation of
protein profiles within complex biological samples.

```{r}
dataSumm <- summarize(dataImput, saveSumm = TRUE)
```

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(dataSumm)
```
</div>

The column "Stat" in the generated result includes the following statistics:

+ n: number.
+ mean: mean.
+ sd: standard deviation.
+ median: median.
+ trimmed: trimmed mean with a trim of 0.1.
+ mad: median absolute deviation (from the median).
+ min: minimum.
+ max: maximum.
+ range: the difference between the maximum and minimum value.
+ skew: skewness.
+ kurtosis: kurtosis.
+ se: standard error.

## Analysis

The function `analyze()` calculates the results that can be used in subsequent
visualizations. If more than two conditions exist in the data, precisely two conditions
for comparison must be specified via the argument `conditions`.

```{r}
cond <- c("50fmol", "100fmol")
```

### Student's t-test

The Student's t-test is used to compare the means between two conditions for each protein,
reporting both the difference in means between the conditions (calculated as Condition 1 -
Condition 2) and the P-value of the test.

```{r}
anlys_t <- analyze(dataImput, conditions = cond, testType = "t-test")
```

Oops! The warning message shows "Data are essentially constant," which means that the data
contain proteins with the same value in all samples. In this case, the P-value of t-test
returns NA.

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(anlys_t)
```
</div>


### Moderated t-test

The main distinction between the Student's and moderated t-tests [@smyth2004linear] lies
in how variance is computed. While the Student's t-test calculates variance based on the
data available for each protein individually, the moderated t-test utilizes information
from all the chosen proteins to calculate variance.

```{r}
anlys_mod.t <- analyze(dataImput, conditions = cond, testType = "mod.t-test")
```

In the moderated t-test, a warning message might occur stating, "Zero sample variances
detected, have been offset away from zero." This warning corresponds to examples of
proteins that exhibited identical quant values, either pre- or post-imputation, and
therefore no variance is present across conditions for those proteins. This does not
impede downstream analysis; it merely serves to alert users to its occurrence.

<!-- This just means that for at least one protein the log ratio is identical for all samples. -->
<!-- Since this will give a zero variance (which will end up in the denominator of your -->
<!-- statistic and could possibly result in an infinite value for your test statistic) it has -->
<!-- been offset to a small value to prevent that possibility. -->

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(anlys_mod.t)
```
</div>


### MA

The result of `testType = "MA"` is to generate the data for plotting an MA plot, which
represents the protein-wise averages within each condition.

```{r}
anlys_MA <- analyze(dataImput, conditions = cond, testType = "MA")
```

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(anlys_MA)
```
</div>


## Visualization

### heatmap

The package offers two options for plotting the heatmap.

+ Option 1 utilizes the source package `pheatmap`, capable of plotting the dendrogram
simultaneously. It is the default choice for heatmaps in this package.

```{r}
visualize(dataImput, graphType = "heatmap",
          pkg = "pheatmap",
          cluster_cols = TRUE, cluster_rows = FALSE,
          show_colnames = TRUE, show_rownames = TRUE)
```

When protein names are excessively long, it is recommended to set `show_rownames = FALSE`
to view the full heatmap.

+ Option 2 use the source package `ggplot2` to generates a ggplot object but does not
include the dendrogram.

```{r}
visualize(dataImput, graphType = "heatmap", pkg = "ggplot2")
```

In a heatmap, similar colors within a row indicate relatively consistent values,
suggesting similar protein expression levels across different samples.


### MA

<!-- An MA plot visualizes the differences between measurements taken in two samples, -->
<!-- by transforming the data onto M (log ratio or fold change) and A (mean average) scales. -->
<!-- The MA plot puts the variable M on the y-axis and A on the x-axis and gives a quick -->
<!-- overview of the distribution of the data. Most proteins are expected to show little -->
<!-- variation, and so the majority of the points will be concentrated around the M = 0 line -->
<!-- (no difference between group means). Typically, points falling above $abs(1)$ are -->
<!-- highlighted. -->

An MA plot, short for "M vs. A plot," which uses two axes:

+ M axis (vertical): Represents the logarithm (usually base 2) of the fold change, or the
ratio of the expression levels, between two conditions. It is calculated as:
$$M = log_2(X/Y) = log_2 X - log_2 Y$$
+ A axis (horizontal): Represents the average intensity of the two conditions, calculated
as:
$$A = \frac{1}{2}log_2(XY) = \frac{1}{2}\left[log_2(X)+log_2(Y)\right]$$

<!-- So, the M value represents the log ratio of protein expression in a treated condition -->
<!-- compared to a control condition, where higher M values indicate greater differences in -->
<!-- expression between the conditions. Meanwhile, the A value helps to account for overall -->
<!-- differences in expression levels and reduce the influence of technical biases. -->
Most proteins are expected to exhibit little variation, leading to the majority of points
concentrating around the line M = 0 (indicating no difference between group means).

```{r}
visualize(anlys_MA, graphType = "MA", M.thres = 1, transformLabel = "Log2")
```

where `M.thres = 1` means the M thresholds are set to −1 and 1. The scatters are split
into three parts: significant up (M > 1), no significant (-1 $\leq$ M $\leq$ 1), and
significant down (M < -1). And `transformLabel = "Log2"` is used to label the title.
Additionally, the warning message "Removed 16 rows containing missing values" indicates
that there are 16 proteins with no significance.


### Normalize

```{r}
visualize(dataNorm, graphType = "normalize")
```


### PCA

Principal component analysis (PCA) is a powerful technique used in data analysis to
simplify and reduce the dimensionality of large datasets. It transforms original variables
into uncorrelated components that capture the maximum variance. By selecting a subset of
these components, PCA projects the data points onto these key directions, enabling
visualization and analysis in a lower-dimensional space. This aids in identifying patterns
and relationships within complex datasets.

In the visualization for `graphType = "PCA_*"`, the arguments `center` and `scale` are
used to center the data to zero mean and scale to unit variance, with default setting at
`TRUE`. The most common error message is "Cannot rescale a constant/zero column to unit
variance." This clearly occurs when columns representing proteins contain only zeros or
have constant values. Typically, there are two ways to address this error: one is to
remove these proteins, and the other is to set `scale = FALSE`.

**NOTE:** data scaling is done to ensure that the scale differences between different
features do not affect the results of PCA. If not scaled, features with larger scales will
dominate the computation of principal components (PCs).

In the case of `dataImput`, two proteins, namely "TEBP_HUMAN" and "T126B_HUMAN,"
have constant values, leading to the error message. We choose to remove these two proteins
in PCA.

```{r}
dataPCA <- dataImput[,colnames(dataImput) != c("TEBP_HUMAN", "T126B_HUMAN")]
```

#### PCA_scree

One way to help identify how many PCs to retain, is to explore a scree plot. The scree
plot shows the eigenvalues of each PC, which represent the proportion of variance
explained by that component.

```{r eval=FALSE}
visualize(dataPCA, graphType = "PCA_scree", center = TRUE, scale = TRUE,
          addlabels = TRUE, choice = "variance", ncp = 10)
visualize(dataPCA, graphType = "PCA_scree", center = TRUE, scale = TRUE,
          addlabels = TRUE, choice = "eigenvalue", ncp = 10)
```

where `choice` specifies the data to be plotted, either `"variance"` or `"eigenvalue"`,
`addlabels = TRUE` adds information labels at the top of bars/points, and `ncp = 10` sets
the number of dimension to be displayed.

```{r echo=FALSE}
library(gridExtra)
plot1 <- visualize(dataPCA, graphType = "PCA_scree", center = TRUE, scale = TRUE,
                   addlabels = TRUE, choice = "variance", ncp = 10)
plot2 <- visualize(dataPCA, graphType = "PCA_scree", center = TRUE, scale = TRUE,
                   addlabels = TRUE, choice = "eigenvalue", ncp = 10)
grid.arrange(plot1, plot2, ncol = 2)
```


#### PCA_ind

The primary PCA plot of individual data visually represents the distribution of individual
observations in a reduced-dimensional space, typically defined by the PCs. The x and y
axes of the PCA plot represent the PCs. Each axis corresponds to a linear combination of
the original variables. Individual data points on the PCA plot represent observations
(e.g., samples) from the original dataset. Points that are close to the origin (0, 0), are
close to the "average" across all protein abundances. If sufficient samples are present,
the plot will also produce a 95% confidence ellipse, as well as a centroid (mean for each
group provided), for each groups (condition) provided.

```{r}
visualize(dataPCA, graphType = "PCA_ind", center = TRUE, scale = TRUE,
          addlabels = TRUE, addEllipses = TRUE, ellipse.level = 0.95)
```


#### PCA_var

This plot will be more useful if your analyses are based on a relatively small number of
proteins. It represents the association, or loading of each protein on the first two PCs.
Longer arrows represents stronger associations. It is important to note that proteins that
are weakly associated with PC1 or PC2 may still be highly correlated with other PCs not
being plotted. Consult the scree plot (and other available methods) to determine the
appropriate number of PCs to investigate. 

```{r}
visualize(dataPCA, graphType = "PCA_var", center = TRUE, scale = TRUE,
          addlabels = TRUE)
```


#### PCA_biplot

The PCA biplot includes individual and variable plots. Again, with a large number of
proteins, this plot can be unwieldy.

```{r}
visualize(dataPCA, graphType = "PCA_biplot", center = TRUE, scale = TRUE,
          addEllipses = TRUE, ellipse.level = 0.95, label = "all")
```


### t-test

The function `visualize()` can be applied to any t-test output. It generates two useful
plots: a histogram of fold changes across the analyzed proteins and a histogram of
P-values. The majority of proteins are expected to show very small change between
conditions, so the fold change histogram will have a peak at around zero. For the P-values,
most P-values are expected to be non-significant (above 0.05). Depending on the strength
of the treatment effect, there may be a peak of p-values near 0.

```{r}
visualize(anlys_mod.t, graphType = "t-test")
```


### Upset

The upset plot is a visual representation that helps display the overlap and intersection
of sets or categories in a dataset. It is particularly useful for illustrating the
presence or absence of elements in combinations of sets.

```{r}
dataSort <- sortcondition(dataSet)
visualize(dataSort, graphType = "Upset")
```

This plot reveals that 18 proteins are in common between 100fmol and 50fmol, while only 1
protein is unique to 100fmol.


### Venn

The Venn plot is another graphical representation of the relationships between sets. Each
circle represents a set, and the overlapping regions show the elements that are shared
between sets.

```{r}
visualize(dataSort, graphType = "Venn",
          show_percentage = TRUE,
          fill_color = c("blue", "yellow", "green", "red"),
          show_universal = FALSE)
```

In the example above, 100fmol and 50fmol groups share 18 proteins. Notably, one protein is
exclusively found in the 100fmol group, while the 50fmol group lacks any unique proteins.


### Volcano

A volcano plot is a graphical representation commonly used in proteomics and genomics to
visualize differential expression analysis results. It is particularly useful for
identifying significant changes in extensive data. It displays two important pieces of
information about differences between conditions in a dataset:

+ Statistical significance (vertical): Represents the negative log10 of the P-value.

+ Fold change (horizontal): Represents the fold change.

```{r}
visualize(anlys_mod.t, graphType = "volcano",
          P.thres = 0.05, logF.thres = 0.6)
```


## Other useful function

To easily extract all quantitative data for a single protein or set of proteins by name as
it was processed throughout the entire **msDiaLogue** workflow, use the function
`pullProteinPath()` and the protein's unique name.

```{r}
ZC11B <- pullProteinPath(proteinName = "ZC11B_HUMAN",
                         dataSetList = list(Initial = dataSet,
                                            Transformed = dataTran,
                                            Normalized = dataNorm,
                                            Imputed = dataImput))
```

<div style="overflow-x: auto;">
```{r echo=FALSE}
knitr::kable(ZC11B)
```
</div>


## Reference {-}

