<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="msDiaLogue">
<title>A Usage Template for the R Package msDiaLogue • msDiaLogue</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="A Usage Template for the R Package msDiaLogue">
<meta property="og:description" content="msDiaLogue">
<meta property="og:image" content="https://github.com/uconn-scs/msDiaLogue/blob/main/man/figure/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">msDiaLogue</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.0.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/usage_template.html">A Usage Template for the R Package msDiaLogue</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/uconn-scs/msDiaLogue/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>A Usage Template for the R Package msDiaLogue</h1>
                        <h4 data-toc-skip class="author"></h4>
<p>Shiying
Xiao<span class="math inline">\(^1\)</span>, Charles Watt<span class="math inline">\(^1\)</span>, Jennifer C. Liddle<span class="math inline">\(^2\)</span>, Jeremy L. Balsbaugh<span class="math inline">\(^2\)</span>, Timothy E. Moore<span class="math inline">\(^3\)</span></p>
                        <h4 data-toc-skip class="author"></h4>
<p><span class="math inline">\(^1\)</span>Department of Statistics, UConn<br><span class="math inline">\(^2\)</span>Proteomics and Metabolomics
Facility, UConn<br><span class="math inline">\(^3\)</span>Statistical Consulting Services,
UConn</p>
            
            <h4 data-toc-skip class="date">2024-04-01</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/uconn-scs/msDiaLogue/blob/HEAD/vignettes/usage_template.Rmd" class="external-link"><code>vignettes/usage_template.Rmd</code></a></small>
      <div class="d-none name"><code>usage_template.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="load-r-package">Load R package<a class="anchor" aria-label="anchor" href="#load-r-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/uconn-scs/msDiaLogue" class="external-link">msDiaLogue</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preprocessing">Preprocessing<a class="anchor" aria-label="anchor" href="#preprocessing"></a>
</h2>
<p>The function <code><a href="../reference/preprocessing.html">preprocessing()</a></code> takes a <code>.csv</code>
file of summarized protein abundances, exported from
<strong>Spectronaut</strong>. The most important columns that need to be
included in this file are: <code>R.Condition</code>,
<code>R.Replicate</code>, <code>PG.ProteinAccessions</code>,
<code>PG.ProteinNames</code>,
<code>PG.NrOfStrippedSequencesIdentified</code>, and
<code>PG.Quantity</code>. This function will reformat the data and
provide functionality for some initial filtering (based on the number of
unique peptides). The steps below describe the functions that happen in
the Preprocessing code.</p>
<p><strong>1.</strong> Loads the raw data</p>
<ul>
<li><p>If the raw data is in a .csv file <a href="https://github.com/uconn-scs/msDiaLogue/blob/main/tests/testData/Toy_Spectronaut_Data.csv" class="external-link">Toy_Spectronaut_Data.csv</a>,
specify the <code>fileName</code> to read the raw data file into
<strong>R</strong>.</p></li>
<li><p>If the raw data is stored as an .RData file <a href="https://github.com/uconn-scs/msDiaLogue/blob/main/tests/testData/Toy_Spectronaut_Data.RData" class="external-link">Toy_Spectronaut_Data.RData</a>,
first load the data file directly, then specify the <code>dataSet</code>
in the function.</p></li>
</ul>
<p><strong>2.</strong> Filters out identified proteins that exhibit
“NaN” quantitative values</p>
<p>NaN, which stands for ‘Not a Number,’ can be found in the PG.Quantity
column for proteins that were identified by MS and MS/MS evidence in the
raw data, but all peptides from that protein lack an associated
integrated peak area or intensity. This usually occurs in low abundance
peptides that exhibit intensities close to the limit of detection
resulting in poor signal-to-noise (S/N) and/or when there is
interference from other co-eluting peptide ions with very similar or
identical m/z values that lead to difficulty in parsing out individual
intensity profiles.</p>
<p><strong>3.</strong> Applies a unique peptides per protein filter</p>
<p>General practice in the proteomics field is to filter out proteins
from which only 1 unique peptide was identified. This adds increased
confidence to results already filtered to a 1% false discovery rate
(FDR), since proteins that are identified with 2 or more peptides are
less likely to be false positives. We recommend filtering out these
protein entries in order to focus on more confident targets in the
identified proteome. However, 1-peptide proteins can still be observed
in the original protein report from <strong>Spectronaut</strong>.</p>
<p><strong>4.</strong> Adds accession numbers to identified proteins
without informative names</p>
<p><strong>Spectronaut</strong> reports contain 4 different columns of
identifying information:</p>
<ul>
<li>
<code>PG.Genes</code>, which is the gene name (e.g. CDK1).</li>
<li>
<code>PG.ProteinAccessions</code>, which is the UniProt identifier
number for a unique entry in the online database (e.g. P06493).</li>
<li>
<code>PG.ProteinDescriptions</code>, which is the protein name as
provided on UniProt (e.g. cyclin-dependent kinase 1).</li>
<li>
<code>PG.ProteinNames</code>, which is a concatenation of an
identifier and the species (e.g. CDK1_HUMAN).</li>
</ul>
<p>Every entry in UniProt will have an accession number, but may not
have all of the other identifiers, due to incomplete annotation. Because
Uniprot includes entries for fragments of proteins and some proteins
entries are redundant, a peptide can match to multiple entries for the
same protein, which generates multiple possible identifiers in
<strong>Spectronaut</strong>. Further, the <code>ProteinNames</code>
entry in <strong>Spectronaut</strong> can switch formats: the preference
is accession number and species, but can also be gene name and species
instead.</p>
<p>This option tells <strong>msDiaLogue</strong> to substitute the
accession number for an identifier if it tries to pull an identifier
from a column with no information.</p>
<p><strong>NOTE:</strong> Not all proteins can be identified
unambiguously. In many cases, the identified peptides can be found in
multiple protein sequences, which yields a protein group or protein
cluster rather than a single protein identification. When this happens,
the accession numbers for all potential matches are concatenated into
one string, separated by periods. When you see long strings of multiple
identifiers later in your data processing, this is why. Spectronaut
sorts these alphanumerically, so you should not assume that the first
protein in the list is most likely to be correct (as is the case in
other search algorithms).</p>
<p><strong>5.</strong> Saves a document to your working directory with
all filtered out data, if desired</p>
<p>If <code>saveRm = TRUE</code>, the data removed in step 2
(<em>preprocess_Filtered_Out_NaN.csv</em>) and step 3
(<em>preprocess_Filtered_Out_Unique.csv</em>) will be saved in the
current working directory.</p>
<p>As part of the <code><a href="../reference/preprocessing.html">preprocessing()</a></code>, a histogram of <span class="math inline">\(log_2\)</span>-transformed protein abundances is
provided. This is a helpful way to confirm that the data have been read
in correctly, and there are no issues with the numerical values of the
protein abundances. Ideally, this histogram will appear fairly
symmetrical (bell-shaped) without too much skew towards smaller or
larger values.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## if the raw data is in a .csv file</span></span>
<span><span class="va">fileName</span> <span class="op">&lt;-</span> <span class="st">"../tests/testData/Toy_Spectronaut_Data.csv"</span></span>
<span><span class="va">dataSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocessing.html">preprocessing</a></span><span class="op">(</span><span class="va">fileName</span>,</span>
<span>                         filterNaN <span class="op">=</span> <span class="cn">TRUE</span>, filterUnique <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                         replaceBlank <span class="op">=</span> <span class="cn">TRUE</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><strong>NOTE:</strong> <code><a href="../reference/preprocessing.html">preprocessing()</a></code> does not perform
a transformation on your data. You still need to use the function
<code><a href="../reference/transform.html">transform()</a></code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## if the raw data is in an .Rdata file</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"../tests/testData/Toy_Spectronaut_Data.RData"</span><span class="op">)</span></span>
<span><span class="va">dataSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/preprocessing.html">preprocessing</a></span><span class="op">(</span>dataSet <span class="op">=</span> <span class="va">Toy_Spectronaut_Data</span>,</span>
<span>                         filterNaN <span class="op">=</span> <span class="cn">TRUE</span>, filterUnique <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                         replaceBlank <span class="op">=</span> <span class="cn">TRUE</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 25 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_bin()`).</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"><img src="usage_template_files/figure-html/unnamed-chunk-4-2.png" width="700" style="display: block; margin: auto;"></p>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="21%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="left">R.Replicate</th>
<th align="left">RAB3D_HUMAN</th>
<th align="left">ADH1_YEAST</th>
<th align="left">LYSC_CHICK</th>
<th align="left">BGAL_HUMAN</th>
<th align="left">SYTC_HUMAN</th>
<th align="left">CYC_BOVIN</th>
<th align="left">PA1B2_HUMAN</th>
<th align="left">TEBP_HUMAN</th>
<th align="left">UAP1_HUMAN</th>
<th align="left">B3GLT_HUMAN</th>
<th align="left">NFXL1_HUMAN</th>
<th align="left">VPS36_HUMAN</th>
<th align="left">T126B_HUMAN</th>
<th align="left">XPO4_HUMAN</th>
<th align="left">ORC3_HUMAN</th>
<th align="left">BAG5_HUMAN</th>
<th align="left">ANGL3_HUMAN</th>
<th align="left">ZC11B_HUMAN</th>
<th align="left">MAP11_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="left">1</td>
<td align="left">105497.85</td>
<td align="left">1254717.6</td>
<td align="left">1060555.9</td>
<td align="left">224187.0</td>
<td align="left">2410569</td>
<td align="left">464875.9</td>
<td align="left">181490.1</td>
<td align="left">4189765</td>
<td align="left">345700.1</td>
<td align="left">24545.98</td>
<td align="left">69218.43</td>
<td align="left">91874.47</td>
<td align="left">13348.120</td>
<td align="left">426354.1</td>
<td align="left">88630.06</td>
<td align="left">227958.8</td>
<td align="left">32128.59</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="left">2</td>
<td align="left">97413.77</td>
<td align="left">1220362.0</td>
<td align="left">1059899.6</td>
<td align="left">227931.1</td>
<td align="left">2396752</td>
<td align="left">477952.2</td>
<td align="left">175121.3</td>
<td align="left">4230263</td>
<td align="left">339977.9</td>
<td align="left">25938.39</td>
<td align="left">69400.23</td>
<td align="left">86165.41</td>
<td align="left">9359.288</td>
<td align="left">395663.8</td>
<td align="left">87888.85</td>
<td align="left">217351.5</td>
<td align="left">31688.79</td>
<td align="left">NA</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="left">3</td>
<td align="left">92411.47</td>
<td align="left">1362763.1</td>
<td align="left">1055651.2</td>
<td align="left">219066.4</td>
<td align="left">2358946</td>
<td align="left">446657.5</td>
<td align="left">181810.5</td>
<td align="left">4384376</td>
<td align="left">326688.5</td>
<td align="left">22291.07</td>
<td align="left">71694.90</td>
<td align="left">91814.37</td>
<td align="left">10157.106</td>
<td align="left">NA</td>
<td align="left">88932.05</td>
<td align="left">202663.6</td>
<td align="left">34704.88</td>
<td align="left">172537.11</td>
<td align="left">489415.6</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="left">4</td>
<td align="left">100762.16</td>
<td align="left">1270573.4</td>
<td align="left">1060255.8</td>
<td align="left">209789.4</td>
<td align="left">2388449</td>
<td align="left">458660.3</td>
<td align="left">165448.0</td>
<td align="left">4272664</td>
<td align="left">333170.9</td>
<td align="left">23725.73</td>
<td align="left">66992.01</td>
<td align="left">99058.58</td>
<td align="left">13830.882</td>
<td align="left">NA</td>
<td align="left">87089.87</td>
<td align="left">230624.2</td>
<td align="left">29539.77</td>
<td align="left">172585.23</td>
<td align="left">507468.9</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="left">5</td>
<td align="left">99035.44</td>
<td align="left">1234380.8</td>
<td align="left">1021856.6</td>
<td align="left">206950.5</td>
<td align="left">2325342</td>
<td align="left">473778.7</td>
<td align="left">173582.3</td>
<td align="left">4066718</td>
<td align="left">333222.6</td>
<td align="left">20868.58</td>
<td align="left">66214.05</td>
<td align="left">94079.65</td>
<td align="left">13272.666</td>
<td align="left">NA</td>
<td align="left">83254.25</td>
<td align="left">224590.7</td>
<td align="left">23633.19</td>
<td align="left">136210.98</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="left">1</td>
<td align="left">90283.59</td>
<td align="left">686584.4</td>
<td align="left">601005.6</td>
<td align="left">293384.3</td>
<td align="left">2866171</td>
<td align="left">251238.0</td>
<td align="left">189993.1</td>
<td align="left">5028592</td>
<td align="left">368101.3</td>
<td align="left">46684.45</td>
<td align="left">66062.98</td>
<td align="left">118289.54</td>
<td align="left">8576.864</td>
<td align="left">NA</td>
<td align="left">114119.27</td>
<td align="left">272555.8</td>
<td align="left">24733.25</td>
<td align="left">60972.31</td>
<td align="left">583439.9</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="left">2</td>
<td align="left">103628.16</td>
<td align="left">764785.9</td>
<td align="left">580436.1</td>
<td align="left">312817.6</td>
<td align="left">3090666</td>
<td align="left">272467.7</td>
<td align="left">211132.2</td>
<td align="left">5039420</td>
<td align="left">412326.3</td>
<td align="left">41758.35</td>
<td align="left">72337.18</td>
<td align="left">116845.31</td>
<td align="left">16022.972</td>
<td align="left">NA</td>
<td align="left">112596.48</td>
<td align="left">305347.0</td>
<td align="left">29952.96</td>
<td align="left">NA</td>
<td align="left">603853.8</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="left">3</td>
<td align="left">102738.70</td>
<td align="left">742963.7</td>
<td align="left">588663.6</td>
<td align="left">323090.6</td>
<td align="left">2981510</td>
<td align="left">261662.1</td>
<td align="left">184879.4</td>
<td align="left">5044124</td>
<td align="left">387863.9</td>
<td align="left">41997.87</td>
<td align="left">71494.94</td>
<td align="left">114235.21</td>
<td align="left">18709.572</td>
<td align="left">NA</td>
<td align="left">111635.11</td>
<td align="left">292004.2</td>
<td align="left">NA</td>
<td align="left">112139.66</td>
<td align="left">605655.4</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="left">4</td>
<td align="left">102062.97</td>
<td align="left">679660.4</td>
<td align="left">585632.6</td>
<td align="left">259369.3</td>
<td align="left">2591955</td>
<td align="left">236597.4</td>
<td align="left">161675.0</td>
<td align="left">4770158</td>
<td align="left">361139.3</td>
<td align="left">32186.93</td>
<td align="left">68405.24</td>
<td align="left">103696.82</td>
<td align="left">10490.210</td>
<td align="left">NA</td>
<td align="left">97793.41</td>
<td align="left">259217.6</td>
<td align="left">24244.19</td>
<td align="left">109480.65</td>
<td align="left">517728.2</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="left">5</td>
<td align="left">100568.80</td>
<td align="left">695747.8</td>
<td align="left">586316.2</td>
<td align="left">272618.9</td>
<td align="left">2624492</td>
<td align="left">246098.1</td>
<td align="left">155629.5</td>
<td align="left">4769110</td>
<td align="left">367231.3</td>
<td align="left">29210.47</td>
<td align="left">67668.77</td>
<td align="left">109244.66</td>
<td align="left">13339.395</td>
<td align="left">NA</td>
<td align="left">95007.62</td>
<td align="left">260105.4</td>
<td align="left">28142.99</td>
<td align="left">NA</td>
<td align="left">541691.9</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="transformation">Transformation<a class="anchor" aria-label="anchor" href="#transformation"></a>
</h2>
<p>Raw mass spectrometry intensity measurements are often unsuitable for
direct statistical modeling because the shape of the data is usually not
symmetrical and the variance is not consistent across the range of
intensities. Most proteomic workflows will convert these raw values with
a log<span class="math inline">\(_2\)</span> transformation, which both
reshapes the data into a more symmetrical distribution, making it easier
to interpret mean-based fold changes, and also stabilizes the variance
across the intensity range (i.e. reduces heteroscedasticity).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataTran</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transform.html">transform</a></span><span class="op">(</span><span class="va">dataSet</span>, logFold <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-6-1.png" width="700" style="display: block; margin: auto;"><img src="usage_template_files/figure-html/unnamed-chunk-6-2.png" width="700" style="display: block; margin: auto;"></p>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="21%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">XPO4_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
<th align="right">MAP11_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.68685</td>
<td align="right">20.25893</td>
<td align="right">20.01639</td>
<td align="right">17.77434</td>
<td align="right">21.20094</td>
<td align="right">18.82649</td>
<td align="right">17.46953</td>
<td align="right">21.99844</td>
<td align="right">18.39916</td>
<td align="right">14.58320</td>
<td align="right">16.07887</td>
<td align="right">16.48738</td>
<td align="right">13.70435</td>
<td align="right">18.70169</td>
<td align="right">16.43551</td>
<td align="right">17.79841</td>
<td align="right">14.97157</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.57184</td>
<td align="right">20.21888</td>
<td align="right">20.01550</td>
<td align="right">17.79824</td>
<td align="right">21.19265</td>
<td align="right">18.86651</td>
<td align="right">17.41799</td>
<td align="right">22.01232</td>
<td align="right">18.37508</td>
<td align="right">14.66280</td>
<td align="right">16.08265</td>
<td align="right">16.39482</td>
<td align="right">13.19218</td>
<td align="right">18.59392</td>
<td align="right">16.42339</td>
<td align="right">17.72967</td>
<td align="right">14.95169</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.49578</td>
<td align="right">20.37810</td>
<td align="right">20.00970</td>
<td align="right">17.74101</td>
<td align="right">21.16971</td>
<td align="right">18.76881</td>
<td align="right">17.47208</td>
<td align="right">22.06394</td>
<td align="right">18.31756</td>
<td align="right">14.44418</td>
<td align="right">16.12958</td>
<td align="right">16.48643</td>
<td align="right">13.31020</td>
<td align="right">NA</td>
<td align="right">16.44042</td>
<td align="right">17.62873</td>
<td align="right">15.08285</td>
<td align="right">17.39655</td>
<td align="right">18.90070</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.62059</td>
<td align="right">20.27705</td>
<td align="right">20.01598</td>
<td align="right">17.67858</td>
<td align="right">21.18764</td>
<td align="right">18.80707</td>
<td align="right">17.33602</td>
<td align="right">22.02670</td>
<td align="right">18.34590</td>
<td align="right">14.53416</td>
<td align="right">16.03170</td>
<td align="right">16.59599</td>
<td align="right">13.75561</td>
<td align="right">NA</td>
<td align="right">16.41022</td>
<td align="right">17.81518</td>
<td align="right">14.85037</td>
<td align="right">17.39695</td>
<td align="right">18.95296</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.59566</td>
<td align="right">20.23536</td>
<td align="right">19.96276</td>
<td align="right">17.65893</td>
<td align="right">21.14901</td>
<td align="right">18.85385</td>
<td align="right">17.40526</td>
<td align="right">21.95543</td>
<td align="right">18.34613</td>
<td align="right">14.34904</td>
<td align="right">16.01485</td>
<td align="right">16.52160</td>
<td align="right">13.69617</td>
<td align="right">NA</td>
<td align="right">16.34524</td>
<td align="right">17.77694</td>
<td align="right">14.52853</td>
<td align="right">17.05548</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.46218</td>
<td align="right">19.38908</td>
<td align="right">19.19702</td>
<td align="right">18.16243</td>
<td align="right">21.45069</td>
<td align="right">17.93870</td>
<td align="right">17.53559</td>
<td align="right">22.26172</td>
<td align="right">18.48974</td>
<td align="right">15.51065</td>
<td align="right">16.01155</td>
<td align="right">16.85196</td>
<td align="right">13.06623</td>
<td align="right">NA</td>
<td align="right">16.80018</td>
<td align="right">18.05619</td>
<td align="right">14.59416</td>
<td align="right">15.89587</td>
<td align="right">19.15422</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.66106</td>
<td align="right">19.54470</td>
<td align="right">19.14678</td>
<td align="right">18.25496</td>
<td align="right">21.55949</td>
<td align="right">18.05573</td>
<td align="right">17.68779</td>
<td align="right">22.26483</td>
<td align="right">18.65343</td>
<td align="right">15.34978</td>
<td align="right">16.14245</td>
<td align="right">16.83424</td>
<td align="right">13.96785</td>
<td align="right">NA</td>
<td align="right">16.78080</td>
<td align="right">18.22009</td>
<td align="right">14.87041</td>
<td align="right">NA</td>
<td align="right">19.20384</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.64862</td>
<td align="right">19.50293</td>
<td align="right">19.16708</td>
<td align="right">18.30158</td>
<td align="right">21.50761</td>
<td align="right">17.99735</td>
<td align="right">17.49622</td>
<td align="right">22.26617</td>
<td align="right">18.56519</td>
<td align="right">15.35803</td>
<td align="right">16.12555</td>
<td align="right">16.80165</td>
<td align="right">14.19149</td>
<td align="right">NA</td>
<td align="right">16.76843</td>
<td align="right">18.15563</td>
<td align="right">NA</td>
<td align="right">16.77494</td>
<td align="right">19.20814</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.63910</td>
<td align="right">19.37445</td>
<td align="right">19.15964</td>
<td align="right">17.98465</td>
<td align="right">21.30561</td>
<td align="right">17.85207</td>
<td align="right">17.30274</td>
<td align="right">22.18561</td>
<td align="right">18.46220</td>
<td align="right">14.97419</td>
<td align="right">16.06182</td>
<td align="right">16.66201</td>
<td align="right">13.35676</td>
<td align="right">NA</td>
<td align="right">16.57745</td>
<td align="right">17.98380</td>
<td align="right">14.56535</td>
<td align="right">16.74032</td>
<td align="right">18.98184</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.61782</td>
<td align="right">19.40820</td>
<td align="right">19.16132</td>
<td align="right">18.05653</td>
<td align="right">21.32361</td>
<td align="right">17.90887</td>
<td align="right">17.24776</td>
<td align="right">22.18529</td>
<td align="right">18.48633</td>
<td align="right">14.83420</td>
<td align="right">16.04620</td>
<td align="right">16.73720</td>
<td align="right">13.70341</td>
<td align="right">NA</td>
<td align="right">16.53576</td>
<td align="right">17.98874</td>
<td align="right">14.78049</td>
<td align="right">NA</td>
<td align="right">19.04711</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="filtering">Filtering<a class="anchor" aria-label="anchor" href="#filtering"></a>
</h2>
<p>In some cases, a researcher may wish to filter out a specific protein
or proteins from the dataset. The most common instance of this would be
proteins identified from the common contaminants database, where the
identification is necessary to avoid incorrect matching but the result
is irrelevant to the experimental question and would not be included in
data visualization. Other scenarios might include a mixed-species
experiment where the researcher wants to evaluate data from only one
species at a time. This step allows you to set aside specific proteins
from downstream analysis, using the gene_species identifier format.</p>
<p><strong>NOTE:</strong> The proteins to be selected or removed is the
union of those specified in <code>listName</code> and those matching the
regular expression pattern in <code>regexName</code>.</p>
<p><strong>KEEP IN MIND:</strong> Removal of any proteins, including
common contaminants, will affect any global calculations performed after
this step (such as normalization). This should not be done without a
clear understanding of how this will affect your results.</p>
<p><strong>Case 1.</strong> Remove proteins specified by the user in
this step and keep everything else.</p>
<p>For example, the proteins named “ADH1_YEAST” and those containing
“HUMAN” are chosen to be filtered out.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterOutIn.html">filterOutIn</a></span><span class="op">(</span><span class="va">dataTran</span>, listName <span class="op">=</span> <span class="st">"ADH1_YEAST"</span>, regexName <span class="op">=</span> <span class="st">"HUMAN"</span>,</span>
<span>            removeList <span class="op">=</span> <span class="cn">TRUE</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>where <code>removeList = TRUE</code> indicates the removal of
proteins from the union of <code>listName</code> and
<code>regexName</code> in <code>dataTran</code>. Please note that if
<code>saveRm = TRUE</code>, the excluded data (“ADH1_YEAS” + “*HUMAN”)
will be saved as a .csv file named <em>filtered_out_data.csv</em> in the
current working directory.</p>
<div style="overflow-x: auto;">
<table class="table">
<colgroup>
<col width="10%">
<col width="59%">
<col width="10%">
<col width="10%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">LYSC_CHICK</th>
<th align="right">CYC_BOVIN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">20.01639</td>
<td align="right">18.82649</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">20.01550</td>
<td align="right">18.86651</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">20.00970</td>
<td align="right">18.76881</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">20.01598</td>
<td align="right">18.80707</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">19.96276</td>
<td align="right">18.85385</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">19.19702</td>
<td align="right">17.93870</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">19.14678</td>
<td align="right">18.05573</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">19.16708</td>
<td align="right">17.99735</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">19.15964</td>
<td align="right">17.85207</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">19.16132</td>
<td align="right">17.90887</td>
</tr>
</tbody>
</table>
</div>
<p><strong>Case 2.</strong> Keep the proteins specified by the user in
this step and remove everything else.</p>
<p>Alternatively, if we would to keep proteins like “ADH1_YEAST” and
“*HUMAN”, simply set <code>removelist = FALSE</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterOutIn.html">filterOutIn</a></span><span class="op">(</span><span class="va">dataTran</span>, listName <span class="op">=</span> <span class="st">"ADH1_YEAST"</span>, regexName <span class="op">=</span> <span class="st">"HUMAN"</span>,</span>
<span>            removeList <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<div style="overflow-x: auto;">
<table class="table">
<colgroup>
<col width="4%">
<col width="22%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">XPO4_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
<th align="right">MAP11_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.68685</td>
<td align="right">20.25893</td>
<td align="right">17.77434</td>
<td align="right">21.20094</td>
<td align="right">17.46953</td>
<td align="right">21.99844</td>
<td align="right">18.39916</td>
<td align="right">14.58320</td>
<td align="right">16.07887</td>
<td align="right">16.48738</td>
<td align="right">13.70435</td>
<td align="right">18.70169</td>
<td align="right">16.43551</td>
<td align="right">17.79841</td>
<td align="right">14.97157</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.57184</td>
<td align="right">20.21888</td>
<td align="right">17.79824</td>
<td align="right">21.19265</td>
<td align="right">17.41799</td>
<td align="right">22.01232</td>
<td align="right">18.37508</td>
<td align="right">14.66280</td>
<td align="right">16.08265</td>
<td align="right">16.39482</td>
<td align="right">13.19218</td>
<td align="right">18.59392</td>
<td align="right">16.42339</td>
<td align="right">17.72967</td>
<td align="right">14.95169</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.49578</td>
<td align="right">20.37810</td>
<td align="right">17.74101</td>
<td align="right">21.16971</td>
<td align="right">17.47208</td>
<td align="right">22.06394</td>
<td align="right">18.31756</td>
<td align="right">14.44418</td>
<td align="right">16.12958</td>
<td align="right">16.48643</td>
<td align="right">13.31020</td>
<td align="right">NA</td>
<td align="right">16.44042</td>
<td align="right">17.62873</td>
<td align="right">15.08285</td>
<td align="right">17.39655</td>
<td align="right">18.90070</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.62059</td>
<td align="right">20.27705</td>
<td align="right">17.67858</td>
<td align="right">21.18764</td>
<td align="right">17.33602</td>
<td align="right">22.02670</td>
<td align="right">18.34590</td>
<td align="right">14.53416</td>
<td align="right">16.03170</td>
<td align="right">16.59599</td>
<td align="right">13.75561</td>
<td align="right">NA</td>
<td align="right">16.41022</td>
<td align="right">17.81518</td>
<td align="right">14.85037</td>
<td align="right">17.39695</td>
<td align="right">18.95296</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.59566</td>
<td align="right">20.23536</td>
<td align="right">17.65893</td>
<td align="right">21.14901</td>
<td align="right">17.40526</td>
<td align="right">21.95543</td>
<td align="right">18.34613</td>
<td align="right">14.34904</td>
<td align="right">16.01485</td>
<td align="right">16.52160</td>
<td align="right">13.69617</td>
<td align="right">NA</td>
<td align="right">16.34524</td>
<td align="right">17.77694</td>
<td align="right">14.52853</td>
<td align="right">17.05548</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.46218</td>
<td align="right">19.38908</td>
<td align="right">18.16243</td>
<td align="right">21.45069</td>
<td align="right">17.53559</td>
<td align="right">22.26172</td>
<td align="right">18.48974</td>
<td align="right">15.51065</td>
<td align="right">16.01155</td>
<td align="right">16.85196</td>
<td align="right">13.06623</td>
<td align="right">NA</td>
<td align="right">16.80018</td>
<td align="right">18.05619</td>
<td align="right">14.59416</td>
<td align="right">15.89587</td>
<td align="right">19.15422</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.66106</td>
<td align="right">19.54470</td>
<td align="right">18.25496</td>
<td align="right">21.55949</td>
<td align="right">17.68779</td>
<td align="right">22.26483</td>
<td align="right">18.65343</td>
<td align="right">15.34978</td>
<td align="right">16.14245</td>
<td align="right">16.83424</td>
<td align="right">13.96785</td>
<td align="right">NA</td>
<td align="right">16.78080</td>
<td align="right">18.22009</td>
<td align="right">14.87041</td>
<td align="right">NA</td>
<td align="right">19.20384</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.64862</td>
<td align="right">19.50293</td>
<td align="right">18.30158</td>
<td align="right">21.50761</td>
<td align="right">17.49622</td>
<td align="right">22.26617</td>
<td align="right">18.56519</td>
<td align="right">15.35803</td>
<td align="right">16.12555</td>
<td align="right">16.80165</td>
<td align="right">14.19149</td>
<td align="right">NA</td>
<td align="right">16.76843</td>
<td align="right">18.15563</td>
<td align="right">NA</td>
<td align="right">16.77494</td>
<td align="right">19.20814</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.63910</td>
<td align="right">19.37445</td>
<td align="right">17.98465</td>
<td align="right">21.30561</td>
<td align="right">17.30274</td>
<td align="right">22.18561</td>
<td align="right">18.46220</td>
<td align="right">14.97419</td>
<td align="right">16.06182</td>
<td align="right">16.66201</td>
<td align="right">13.35676</td>
<td align="right">NA</td>
<td align="right">16.57745</td>
<td align="right">17.98380</td>
<td align="right">14.56535</td>
<td align="right">16.74032</td>
<td align="right">18.98184</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.61782</td>
<td align="right">19.40820</td>
<td align="right">18.05653</td>
<td align="right">21.32361</td>
<td align="right">17.24776</td>
<td align="right">22.18529</td>
<td align="right">18.48633</td>
<td align="right">14.83420</td>
<td align="right">16.04620</td>
<td align="right">16.73720</td>
<td align="right">13.70341</td>
<td align="right">NA</td>
<td align="right">16.53576</td>
<td align="right">17.98874</td>
<td align="right">14.78049</td>
<td align="right">NA</td>
<td align="right">19.04711</td>
</tr>
</tbody>
</table>
</div>
<p><strong>Extension</strong></p>
<p>Besides protein names, the function <code><a href="../reference/filterProtein.html">filterProtein()</a></code>
provides a similar function to filter proteins by additional protein
information: gene, accession, and description.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/filterProtein.html">filterProtein</a></span><span class="op">(</span><span class="va">dataTran</span>, proteinInformation <span class="op">=</span> <span class="st">"preprocess_protein_information.csv"</span>,</span>
<span>              text <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ras-related protein Rab-3D"</span>, <span class="st">"Alcohol dehydrogenase 1"</span><span class="op">)</span>,</span>
<span>              by <span class="op">=</span> <span class="st">"description"</span>,</span>
<span>              removeList <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>where <code>proteinInformation</code> is the file name for protein
information, automatically generated by <code><a href="../reference/preprocessing.html">preprocessing()</a></code>. In
this case, the proteins with descriptions “Ras-related protein Rab-3D”
or “Alcohol dehydrogenase 1” will be kept. Note that the search value
<code>text</code> is used for exact equality search.</p>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="10%">
<col width="58%">
<col width="10%">
<col width="10%">
<col width="9%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.68685</td>
<td align="right">20.25893</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.57184</td>
<td align="right">20.21888</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.49578</td>
<td align="right">20.37810</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.62059</td>
<td align="right">20.27705</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.59566</td>
<td align="right">20.23536</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.46218</td>
<td align="right">19.38908</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.66106</td>
<td align="right">19.54470</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.64862</td>
<td align="right">19.50293</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.63910</td>
<td align="right">19.37445</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.61782</td>
<td align="right">19.40820</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="normalization">Normalization<a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<p>Normalization is designed to address systematic biases in the data.
Biases can arise from inadvertent sample grouping during generation or
preparation, from variations in instrument performance during
acquisition, analysis of different peptide amounts across experiments,
or other reasons. These factors can artificially mask or enhance actual
biological changes.</p>
<p>Many normalization methods have been developed for large datasets,
each with its own strengths and weaknesses. The following factors should
be considered when choosing a normalization method:</p>
<ol style="list-style-type: lower-alpha">
<li><p>Experiment-Specific Normalization:<br>
Most experiments run with <a href="https://proteomics.uconn.edu" class="external-link">UConn
PMF</a> are normalized by injection amount at the time of analysis to
facilitate comparison. “Amount” is measured by UV absorbance at 280 nm,
a standard method for generic protein quantification.</p></li>
<li><p>Assumption of Non-Changing Species:<br>
Most biological experiments implicitly assume that the majority of
measured species in an experiment will not change across conditions.
This assumption is more robust if there are thousands of species,
compared to only hundreds, or tens, so for experiments of very different
complexities (e.g. a purified protein vs. an immunoprecipitation vs. a
full lysate), normalization should not be applied as a global process,
but instead only on subsets of experiments that are relatively similar
to each other.</p></li>
</ol>
<p>So far, this package provides three normalization methods for
use:</p>
<ol style="list-style-type: decimal">
<li><p>“quant”: Quantile <span class="citation">(Bolstad et al.
2003)</span> (values in each run are ranked, quantile bins are applied
to the entire dataset, and values in each run adjusted to their closest
bin value)</p></li>
<li><p>“median”: Protein-wise Median (a scalar factor is applied to each
protein entry to make the median of each sample equal to every other
sample)</p></li>
<li><p>“mean”: Protein-wise Mean (a scalar factor is applied to each
protein entry to make the mean of each sample equal to every other
sample)</p></li>
</ol>
<p>Quantile normalization is generally recommended by <a href="https://statsconsulting.uconn.edu" class="external-link">UConn SCS</a>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataNorm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">dataTran</span>, normalizeType <span class="op">=</span> <span class="st">"quant"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 16 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-14-1.png" width="700" style="display: block; margin: auto;"></p>
<p>Oh! The message “Warning: Removed 16 rows containing non-finite
values” indicates the presence of 16 NA (Not Available) values in the
data. These NA values arise when a protein was not identified in a
particular sample or condition and are automatically excluded when
generating the boxplot but retained in the actual dataset.</p>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="21%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">XPO4_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
<th align="right">MAP11_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.72739</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.16677</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.59092</td>
<td align="right">13.59442</td>
<td align="right">18.53901</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">15.24429</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.91104</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.16677</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.43331</td>
<td align="right">13.59442</td>
<td align="right">18.53901</td>
<td align="right">16.59092</td>
<td align="right">17.72739</td>
<td align="right">15.24429</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.71547</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.00523</td>
<td align="right">21.30239</td>
<td align="right">18.66148</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.29036</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">16.55997</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.38081</td>
<td align="right">17.81117</td>
<td align="right">15.13860</td>
<td align="right">17.08332</td>
<td align="right">19.10318</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.71547</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">17.81117</td>
<td align="right">21.30239</td>
<td align="right">18.66148</td>
<td align="right">17.08332</td>
<td align="right">22.12204</td>
<td align="right">18.29036</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">16.55997</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.38081</td>
<td align="right">18.00523</td>
<td align="right">15.13860</td>
<td align="right">17.54057</td>
<td align="right">19.10318</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.91104</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.72739</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.59092</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.43331</td>
<td align="right">18.16677</td>
<td align="right">15.24429</td>
<td align="right">17.31263</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.55997</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.29036</td>
<td align="right">21.30239</td>
<td align="right">17.81117</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.66148</td>
<td align="right">15.13860</td>
<td align="right">16.38081</td>
<td align="right">17.08332</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.71547</td>
<td align="right">18.00523</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">19.10318</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.43331</td>
<td align="right">20.00402</td>
<td align="right">18.98797</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">15.24429</td>
<td align="right">16.02729</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.59092</td>
<td align="right">17.91104</td>
<td align="right">14.63812</td>
<td align="right">NA</td>
<td align="right">19.49314</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.02729</td>
<td align="right">20.00402</td>
<td align="right">18.98797</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">14.63812</td>
<td align="right">15.24429</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">NA</td>
<td align="right">16.59092</td>
<td align="right">19.49314</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.55997</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.29036</td>
<td align="right">21.30239</td>
<td align="right">17.81117</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.66148</td>
<td align="right">15.13860</td>
<td align="right">15.92981</td>
<td align="right">16.71547</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.38081</td>
<td align="right">18.00523</td>
<td align="right">14.60737</td>
<td align="right">17.08332</td>
<td align="right">19.10318</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.59092</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">15.24429</td>
<td align="right">16.02729</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">14.63812</td>
<td align="right">NA</td>
<td align="right">18.98797</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="imputation">Imputation<a class="anchor" aria-label="anchor" href="#imputation"></a>
</h2>
<p>The two primary MS/MS acquisition types implemented in large scale
MS-based proteomics have unique advantages and disadvantages.
Traditional Data-Dependent Acquisition (DDA) methods favor specificity
in MS/MS sampling over comprehensive proteome coverage. Small peptide
isolation windows (&lt;3 m/z) result in MS/MS spectra that contain
fragmentation data from ideally only one peptide. This specificity
promotes clear peptide identifications but comes at the expense of added
scan time. In DDA experiments, the number of peptides that can be
selected for MS/MS is limited by instrument scan speeds and is therefore
prioritized by highest peptide abundance. Low abundance peptides are
sampled less frequently for MS/MS and this can result in variable
peptide coverage and many missing protein data across large sample
datasets.</p>
<p>Data-Independent Acquisition (DIA) methods promote comprehensive
peptide coverage over specificity by sampling many peptides for MS/MS
simultaneously. Sequential and large mass isolation windows (4-50 m/z)
are used to isolate large numbers of peptides at once for concurrent
MS/MS. This produces complicated fragmentation spectra, but these
spectra contain data on every observable peptide. A major disadvantage
with this type of acquisition is that DIA MS/MS spectra are incredibly
complex and difficult to deconvolve. Powerful and relatively new
software programs like Spectronaut are capable of successfully parsing
out which fragment ions came from each co-fragmented peptide using
custom libraries, machine learning algorithms, and precisely determined
retention times or measured ion mobility data. Because all observable
ions are sampled for MS/MS, DIA reduces missingness substantially
compared to DDA, though not entirely.</p>
<p>Function <code><a href="../reference/dataMissing.html">dataMissing()</a></code> is designed to summarize the
missingness for each protein, where <code>plot = TRUE</code> indicates
plotting the missingness, and <code>show_labels = TRUE</code> means that
the protein names are displayed in the printed plot. Note that the
visual representation is not generated by default, and the plot
generation time varies with project size.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataMissing</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/dataMissing.html">dataMissing</a></span><span class="op">(</span><span class="va">dataNorm</span>, plot <span class="op">=</span> <span class="cn">TRUE</span>, show_labels <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-16-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The percentage in the protein labels represents the proportion of
missing data in the samples for that protein. For instance, the label
“XPO4_HUMAN (80%)” indicates that, within all observations for the
protein “XPO4_HUMAN”, 80% of the data is missing. Additionally, the
percentage in the legend represents the proportion of missing data in
the whole dataset. In this case, 8.4% of the data in
<code>dataNorm</code> is missing.</p>
<p>Regardless of plot generation, the function
<code><a href="../reference/dataMissing.html">dataMissing()</a></code> always returns a table providing the
following information:</p>
<ul>
<li><p><code>count_miss</code>: The count of missing values for each
protein.</p></li>
<li><p><code>pct-miss</code>: The percentage of missing values for each
protein.</p></li>
<li><p><code>pct_total_miss</code>: The percentage of missing values for
each protein relative to the total missing values in the entire
dataset.</p></li>
</ul>
<div style="overflow-x: auto;">
<table class="table">
<colgroup>
<col width="6%">
<col width="5%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="5%">
<col width="4%">
<col width="4%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="5%">
<col width="5%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">XPO4_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
<th align="right">MAP11_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">count_miss</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">8</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1.00</td>
<td align="right">4</td>
<td align="right">3.00</td>
</tr>
<tr class="even">
<td align="left">pct_miss</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">80</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">10.00</td>
<td align="right">40</td>
<td align="right">30.00</td>
</tr>
<tr class="odd">
<td align="left">pct_total_miss</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">50</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">6.25</td>
<td align="right">25</td>
<td align="right">18.75</td>
</tr>
</tbody>
</table>
</div>
<p>For example, in the case of the protein “XPO4_HUMAN,” there are 8 NA
values in the samples, representing 80% of the missing data for
“XPO4_HUMAN” within that sample and 50% of the total missing data in the
entire dataset.</p>
<p>Various imputation methods have been developed to address the
missing-value issue and assign a reasonable guess of quantitative value
to proteins with missing values. So far, this package provides 10
imputation methods for use:</p>
<ol style="list-style-type: decimal">
<li><p><code><a href="../reference/impute.min_local.html">impute.min_local()</a></code>: Replaces missing values with the
lowest measured value for that protein in that condition.</p></li>
<li><p><code><a href="../reference/impute.min_global.html">impute.min_global()</a></code>: Replaces missing values with
the lowest measured value from any protein found within the entire
dataset.</p></li>
<li><p><code><a href="../reference/impute.knn.html">impute.knn()</a></code>: Replaces missing values using the
k-nearest neighbors algorithm <span class="citation">(Troyanskaya et al.
2001)</span>.</p></li>
<li><p><code><a href="../reference/impute.knn_seq.html">impute.knn_seq()</a></code>: Replaces missing values using the
sequential k-nearest neighbors algorithm <span class="citation">(Kim,
Kim, and Yi 2004)</span>.</p></li>
<li><p><code><a href="../reference/impute.knn_trunc.html">impute.knn_trunc()</a></code>: Replaces missing values using
the truncated k-nearest neighbors algorithm <span class="citation">(Shah
et al. 2017)</span>.</p></li>
<li><p><code><a href="../reference/impute.nuc_norm.html">impute.nuc_norm()</a></code>: Replaces missing values using the
nuclear-norm regularization <span class="citation">(Hastie et al.
2015)</span>.</p></li>
<li><p><code><a href="../reference/impute.mice_cart.html">impute.mice_cart()</a></code>: Replaces missing values using
the classification and regression trees <span class="citation">(Breiman
et al. 1984; Doove, van Buuren, and Dusseldorp 2014; van Buuren
2018)</span>.</p></li>
<li><p><code><a href="../reference/impute.mice_norm.html">impute.mice_norm()</a></code>: Replaces missing values using
the Bayesian linear regression <span class="citation">(Rubin 1987;
Schafer 1997; van Buuren and Groothuis-Oudshoorn 2011)</span>.</p></li>
<li><p><code><a href="../reference/impute.pca_bayes.html">impute.pca_bayes()</a></code>: Replaces missing values using
the Bayesian principal components analysis <span class="citation">(Oba
et al. 2003)</span>.</p></li>
<li><p><code><a href="../reference/impute.pca_prob.html">impute.pca_prob()</a></code>: Replaces missing values using the
probabilistic principal components analysis <span class="citation">(Stacklies et al. 2007)</span>.</p></li>
</ol>
<p>Additional methods will be added later.</p>
<p>For example, to impute the NA value of <code>dataNorm</code> using
<code><a href="../reference/impute.min_local.html">impute.min_local()</a></code>, set the required percentage of values
that must be present in a given protein by condition combination for
values to be imputed to 51%.</p>
<p><strong>NOTE:</strong> There is no rule in the field of proteomics
for filtering based on percentage of missingness, similar to there being
no rule for the number of replicates required to draw a conclusion.
However, reproducible observations make conclusions more credible.
Setting the <code>reqPercentPresent</code> to 0.51 requires that any
protein be observed in a majority of the replicates by condition in
order to be considered. For 3 replicates, this would require 2
measurements to allow imputation of the 3rd value. If only 1 measurement
is seen, the other values will remain NA, and will be filtered out in a
subsequent step.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataImput</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute.min_local.html">impute.min_local</a></span><span class="op">(</span><span class="va">dataNorm</span>, reportImputing <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                              reqPercentPresent <span class="op">=</span> <span class="fl">0.51</span><span class="op">)</span></span></code></pre></div>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="21%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">XPO4_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
<th align="right">MAP11_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.72739</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.16677</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.59092</td>
<td align="right">13.59442</td>
<td align="right">18.53901</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">15.24429</td>
<td align="right">17.08332</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.91104</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.16677</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.43331</td>
<td align="right">13.59442</td>
<td align="right">18.53901</td>
<td align="right">16.59092</td>
<td align="right">17.72739</td>
<td align="right">15.24429</td>
<td align="right">17.08332</td>
<td align="right">NA</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.71547</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.00523</td>
<td align="right">21.30239</td>
<td align="right">18.66148</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.29036</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">16.55997</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.38081</td>
<td align="right">17.81117</td>
<td align="right">15.13860</td>
<td align="right">17.08332</td>
<td align="right">19.10318</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.71547</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">17.81117</td>
<td align="right">21.30239</td>
<td align="right">18.66148</td>
<td align="right">17.08332</td>
<td align="right">22.12204</td>
<td align="right">18.29036</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">16.55997</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.38081</td>
<td align="right">18.00523</td>
<td align="right">15.13860</td>
<td align="right">17.54057</td>
<td align="right">19.10318</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.91104</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.72739</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.59092</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.43331</td>
<td align="right">18.16677</td>
<td align="right">15.24429</td>
<td align="right">17.31263</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.55997</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.29036</td>
<td align="right">21.30239</td>
<td align="right">17.81117</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.66148</td>
<td align="right">15.13860</td>
<td align="right">16.38081</td>
<td align="right">17.08332</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.71547</td>
<td align="right">18.00523</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">19.10318</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.43331</td>
<td align="right">20.00402</td>
<td align="right">18.98797</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">15.24429</td>
<td align="right">16.02729</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.59092</td>
<td align="right">17.91104</td>
<td align="right">14.63812</td>
<td align="right">15.92981</td>
<td align="right">19.49314</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.02729</td>
<td align="right">20.00402</td>
<td align="right">18.98797</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">14.63812</td>
<td align="right">15.24429</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">14.60737</td>
<td align="right">16.59092</td>
<td align="right">19.49314</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.55997</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.29036</td>
<td align="right">21.30239</td>
<td align="right">17.81117</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.66148</td>
<td align="right">15.13860</td>
<td align="right">15.92981</td>
<td align="right">16.71547</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.38081</td>
<td align="right">18.00523</td>
<td align="right">14.60737</td>
<td align="right">17.08332</td>
<td align="right">19.10318</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.59092</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">15.24429</td>
<td align="right">16.02729</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">NA</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">14.63812</td>
<td align="right">15.92981</td>
<td align="right">18.98797</td>
</tr>
</tbody>
</table>
</div>
<p>If <code>reportImputing = TRUE</code>, the returned result structure
will be altered to a list, adding a shadow data frame with imputed data
labels, where 1 indicates the corresponding entries have been imputed,
and 0 indicates otherwise.</p>
<p>After the above imputation, any entries that did not pass the percent
present threshold will still have NA values and will need to be filtered
out.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataImput</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterNA.html">filterNA</a></span><span class="op">(</span><span class="va">dataImput</span>, saveRm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>where <code>saveRm = TRUE</code> indicates that the filtered data
will be saved as a .csv file named <em>filtered_NA_data.csv</em> in the
current working directory.</p>
<p>The <code>dataImput</code> is as follows:</p>
<div style="overflow-x: auto;">
<table class="table">
<colgroup>
<col width="4%">
<col width="22%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="4%">
<col width="3%">
<col width="3%">
<col width="4%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.72739</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.16677</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.59092</td>
<td align="right">13.59442</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">15.24429</td>
<td align="right">17.08332</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.91104</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.16677</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.43331</td>
<td align="right">13.59442</td>
<td align="right">16.59092</td>
<td align="right">17.72739</td>
<td align="right">15.24429</td>
<td align="right">17.08332</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.71547</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.00523</td>
<td align="right">21.30239</td>
<td align="right">18.66148</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.29036</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">16.55997</td>
<td align="right">13.59442</td>
<td align="right">16.38081</td>
<td align="right">17.81117</td>
<td align="right">15.13860</td>
<td align="right">17.08332</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.71547</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">17.81117</td>
<td align="right">21.30239</td>
<td align="right">18.66148</td>
<td align="right">17.08332</td>
<td align="right">22.12204</td>
<td align="right">18.29036</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
<td align="right">16.55997</td>
<td align="right">13.59442</td>
<td align="right">16.38081</td>
<td align="right">18.00523</td>
<td align="right">15.13860</td>
<td align="right">17.54057</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.81611</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">17.91104</td>
<td align="right">21.21903</td>
<td align="right">18.98797</td>
<td align="right">17.72739</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">14.63812</td>
<td align="right">16.02729</td>
<td align="right">16.59092</td>
<td align="right">13.59442</td>
<td align="right">16.43331</td>
<td align="right">18.16677</td>
<td align="right">15.24429</td>
<td align="right">17.31263</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">16.55997</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.29036</td>
<td align="right">21.30239</td>
<td align="right">17.81117</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.66148</td>
<td align="right">15.13860</td>
<td align="right">16.38081</td>
<td align="right">17.08332</td>
<td align="right">13.59442</td>
<td align="right">16.71547</td>
<td align="right">18.00523</td>
<td align="right">14.60737</td>
<td align="right">15.92981</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">16.43331</td>
<td align="right">20.00402</td>
<td align="right">18.98797</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">15.24429</td>
<td align="right">16.02729</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">16.59092</td>
<td align="right">17.91104</td>
<td align="right">14.63812</td>
<td align="right">15.92981</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">16.02729</td>
<td align="right">20.00402</td>
<td align="right">18.98797</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">14.63812</td>
<td align="right">15.24429</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">14.60737</td>
<td align="right">16.59092</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">16.55997</td>
<td align="right">20.06373</td>
<td align="right">19.58448</td>
<td align="right">18.29036</td>
<td align="right">21.30239</td>
<td align="right">17.81117</td>
<td align="right">17.54057</td>
<td align="right">22.12204</td>
<td align="right">18.66148</td>
<td align="right">15.13860</td>
<td align="right">15.92981</td>
<td align="right">16.71547</td>
<td align="right">13.59442</td>
<td align="right">16.38081</td>
<td align="right">18.00523</td>
<td align="right">14.60737</td>
<td align="right">17.08332</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">16.59092</td>
<td align="right">20.00402</td>
<td align="right">19.49314</td>
<td align="right">18.16677</td>
<td align="right">21.21903</td>
<td align="right">17.72739</td>
<td align="right">17.31263</td>
<td align="right">22.12204</td>
<td align="right">18.53901</td>
<td align="right">15.24429</td>
<td align="right">16.02729</td>
<td align="right">16.81611</td>
<td align="right">13.59442</td>
<td align="right">16.43331</td>
<td align="right">17.91104</td>
<td align="right">14.63812</td>
<td align="right">15.92981</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="summarization">Summarization<a class="anchor" aria-label="anchor" href="#summarization"></a>
</h2>
<p>This summarization provides a table of values for each protein in the
final dataset that include the final processed abundances and fold
changes in each condition, and that protein’s statistical relation to
the global dataset in terms of its mean, median, standard deviation, and
other parameters.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataSumm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/summarize.html">summarize</a></span><span class="op">(</span><span class="va">dataImput</span>, saveSumm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div style="overflow-x: auto;">
<table class="table">
<colgroup>
<col width="5%">
<col width="4%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">Condition</th>
<th align="left">Stat</th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">n</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.00000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.00000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">mean</td>
<td align="right">16.7758540</td>
<td align="right">20.0279072</td>
<td align="right">19.5296768</td>
<td align="right">17.8731761</td>
<td align="right">21.2523732</td>
<td align="right">18.8573750</td>
<td align="right">17.3953072</td>
<td align="right">22.12204</td>
<td align="right">18.2906540</td>
<td align="right">14.6258172</td>
<td align="right">15.9882985</td>
<td align="right">16.5470182</td>
<td align="right">13.59442</td>
<td align="right">16.4438325</td>
<td align="right">17.9243208</td>
<td align="right">15.2020139</td>
<td align="right">17.2206289</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">sd</td>
<td align="right">0.0551255</td>
<td align="right">0.0327043</td>
<td align="right">0.0500279</td>
<td align="right">0.1065418</td>
<td align="right">0.0456588</td>
<td align="right">0.1788241</td>
<td align="right">0.2461674</td>
<td align="right">0.00000</td>
<td align="right">0.1519689</td>
<td align="right">0.0168409</td>
<td align="right">0.0533895</td>
<td align="right">0.0654245</td>
<td align="right">0.00000</td>
<td align="right">0.0863134</td>
<td align="right">0.1710908</td>
<td align="right">0.0578860</td>
<td align="right">0.2045679</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">median</td>
<td align="right">16.8161120</td>
<td align="right">20.0040234</td>
<td align="right">19.4931415</td>
<td align="right">17.9110449</td>
<td align="right">21.2190287</td>
<td align="right">18.9879697</td>
<td align="right">17.3126283</td>
<td align="right">22.12204</td>
<td align="right">18.2903590</td>
<td align="right">14.6381160</td>
<td align="right">16.0272887</td>
<td align="right">16.5599694</td>
<td align="right">13.59442</td>
<td align="right">16.4333052</td>
<td align="right">17.9110449</td>
<td align="right">15.2442878</td>
<td align="right">17.0833151</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">trimmed</td>
<td align="right">16.7758540</td>
<td align="right">20.0279072</td>
<td align="right">19.5296768</td>
<td align="right">17.8731761</td>
<td align="right">21.2523732</td>
<td align="right">18.8573750</td>
<td align="right">17.3953072</td>
<td align="right">22.12204</td>
<td align="right">18.2906540</td>
<td align="right">14.6258172</td>
<td align="right">15.9882985</td>
<td align="right">16.5470182</td>
<td align="right">13.59442</td>
<td align="right">16.4438325</td>
<td align="right">17.9243208</td>
<td align="right">15.2020139</td>
<td align="right">17.2206289</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">mad</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.1396408</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.3379478</td>
<td align="right">0.00000</td>
<td align="right">0.1832349</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0458926</td>
<td align="right">0.00000</td>
<td align="right">0.0778229</td>
<td align="right">0.1480813</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">min</td>
<td align="right">16.7154670</td>
<td align="right">20.0040234</td>
<td align="right">19.4931415</td>
<td align="right">17.7273935</td>
<td align="right">21.2190287</td>
<td align="right">18.6614830</td>
<td align="right">17.0833151</td>
<td align="right">22.12204</td>
<td align="right">18.1667688</td>
<td align="right">14.6073689</td>
<td align="right">15.9298133</td>
<td align="right">16.4333052</td>
<td align="right">13.59442</td>
<td align="right">16.3808143</td>
<td align="right">17.7273935</td>
<td align="right">15.1386030</td>
<td align="right">17.0833151</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">max</td>
<td align="right">16.8161120</td>
<td align="right">20.0637329</td>
<td align="right">19.5844796</td>
<td align="right">18.0052314</td>
<td align="right">21.3023900</td>
<td align="right">18.9879697</td>
<td align="right">17.7273935</td>
<td align="right">22.12204</td>
<td align="right">18.5390145</td>
<td align="right">14.6381160</td>
<td align="right">16.0272887</td>
<td align="right">16.5909235</td>
<td align="right">13.59442</td>
<td align="right">16.5909235</td>
<td align="right">18.1667688</td>
<td align="right">15.2442878</td>
<td align="right">17.5405709</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">range</td>
<td align="right">0.1006450</td>
<td align="right">0.0597096</td>
<td align="right">0.0913381</td>
<td align="right">0.2778380</td>
<td align="right">0.0833612</td>
<td align="right">0.3264867</td>
<td align="right">0.6440784</td>
<td align="right">0.00000</td>
<td align="right">0.3722458</td>
<td align="right">0.0307471</td>
<td align="right">0.0974754</td>
<td align="right">0.1576183</td>
<td align="right">0.00000</td>
<td align="right">0.2101091</td>
<td align="right">0.4393753</td>
<td align="right">0.1056848</td>
<td align="right">0.4572559</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">skew</td>
<td align="right">-0.2921187</td>
<td align="right">0.2921187</td>
<td align="right">0.2921187</td>
<td align="right">-0.1530099</td>
<td align="right">0.2921187</td>
<td align="right">-0.2921187</td>
<td align="right">0.1097890</td>
<td align="right">NaN</td>
<td align="right">0.6563002</td>
<td align="right">-0.2921187</td>
<td align="right">-0.2921187</td>
<td align="right">-0.9261273</td>
<td align="right">NaN</td>
<td align="right">0.8334126</td>
<td align="right">0.2273441</td>
<td align="right">-0.2921187</td>
<td align="right">0.6018551</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">kurtosis</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">-1.7975401</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">-1.7922330</td>
<td align="right">NaN</td>
<td align="right">-1.3966206</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">-1.0930665</td>
<td align="right">NaN</td>
<td align="right">-1.1994567</td>
<td align="right">-1.7942036</td>
<td align="right">-2.2533333</td>
<td align="right">-1.6733699</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">se</td>
<td align="right">0.0246529</td>
<td align="right">0.0146258</td>
<td align="right">0.0223732</td>
<td align="right">0.0476469</td>
<td align="right">0.0204192</td>
<td align="right">0.0799726</td>
<td align="right">0.1100894</td>
<td align="right">0.00000</td>
<td align="right">0.0679626</td>
<td align="right">0.0075315</td>
<td align="right">0.0238765</td>
<td align="right">0.0292587</td>
<td align="right">0.00000</td>
<td align="right">0.0386005</td>
<td align="right">0.0765141</td>
<td align="right">0.0258874</td>
<td align="right">0.0914855</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">n</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.00000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.00000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
<td align="right">5.0000000</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">mean</td>
<td align="right">16.4342912</td>
<td align="right">20.0279072</td>
<td align="right">19.3276080</td>
<td align="right">18.2162049</td>
<td align="right">21.2523732</td>
<td align="right">17.7609023</td>
<td align="right">17.4038053</td>
<td align="right">22.12204</td>
<td align="right">18.5880019</td>
<td align="right">15.0807795</td>
<td align="right">15.9218986</td>
<td align="right">16.8494236</td>
<td align="right">13.59442</td>
<td align="right">16.5107630</td>
<td align="right">17.9487195</td>
<td align="right">14.6196677</td>
<td align="right">16.2927357</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">sd</td>
<td align="right">0.2354653</td>
<td align="right">0.0327043</td>
<td align="right">0.3122802</td>
<td align="right">0.0676932</td>
<td align="right">0.0456588</td>
<td align="right">0.0458838</td>
<td align="right">0.1248493</td>
<td align="right">0.00000</td>
<td align="right">0.0670787</td>
<td align="right">0.2530356</td>
<td align="right">0.4159438</td>
<td align="right">0.1378210</td>
<td align="right">0.00000</td>
<td align="right">0.1389384</td>
<td align="right">0.0515880</td>
<td align="right">0.0168409</td>
<td align="right">0.5265619</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">median</td>
<td align="right">16.5599694</td>
<td align="right">20.0040234</td>
<td align="right">19.4931415</td>
<td align="right">18.1667688</td>
<td align="right">21.2190287</td>
<td align="right">17.7273935</td>
<td align="right">17.3126283</td>
<td align="right">22.12204</td>
<td align="right">18.5390145</td>
<td align="right">15.1386030</td>
<td align="right">16.0272887</td>
<td align="right">16.8161120</td>
<td align="right">13.59442</td>
<td align="right">16.4333052</td>
<td align="right">17.9110449</td>
<td align="right">14.6073689</td>
<td align="right">15.9298133</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">trimmed</td>
<td align="right">16.4342912</td>
<td align="right">20.0279072</td>
<td align="right">19.3276080</td>
<td align="right">18.2162049</td>
<td align="right">21.2523732</td>
<td align="right">17.7609023</td>
<td align="right">17.4038053</td>
<td align="right">22.12204</td>
<td align="right">18.5880019</td>
<td align="right">15.0807795</td>
<td align="right">15.9218986</td>
<td align="right">16.8494236</td>
<td align="right">13.59442</td>
<td align="right">16.5107630</td>
<td align="right">17.9487195</td>
<td align="right">14.6196677</td>
<td align="right">16.2927357</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">mad</td>
<td align="right">0.0458926</td>
<td align="right">0.0000000</td>
<td align="right">0.1354179</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.00000</td>
<td align="right">0.0000000</td>
<td align="right">0.1566883</td>
<td align="right">0.1445171</td>
<td align="right">0.0000000</td>
<td align="right">0.00000</td>
<td align="right">0.0778229</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">min</td>
<td align="right">16.0272887</td>
<td align="right">20.0040234</td>
<td align="right">18.9879697</td>
<td align="right">18.1667688</td>
<td align="right">21.2190287</td>
<td align="right">17.7273935</td>
<td align="right">17.3126283</td>
<td align="right">22.12204</td>
<td align="right">18.5390145</td>
<td align="right">14.6381160</td>
<td align="right">15.2442878</td>
<td align="right">16.7154670</td>
<td align="right">13.59442</td>
<td align="right">16.3808143</td>
<td align="right">17.9110449</td>
<td align="right">14.6073689</td>
<td align="right">15.9298133</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">max</td>
<td align="right">16.5909235</td>
<td align="right">20.0637329</td>
<td align="right">19.5844796</td>
<td align="right">18.2903590</td>
<td align="right">21.3023900</td>
<td align="right">17.8111655</td>
<td align="right">17.5405709</td>
<td align="right">22.12204</td>
<td align="right">18.6614830</td>
<td align="right">15.2442878</td>
<td align="right">16.3808143</td>
<td align="right">17.0833151</td>
<td align="right">13.59442</td>
<td align="right">16.7154670</td>
<td align="right">18.0052314</td>
<td align="right">14.6381160</td>
<td align="right">17.0833151</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">range</td>
<td align="right">0.5636348</td>
<td align="right">0.0597096</td>
<td align="right">0.5965099</td>
<td align="right">0.1235902</td>
<td align="right">0.0833612</td>
<td align="right">0.0837721</td>
<td align="right">0.2279426</td>
<td align="right">0.00000</td>
<td align="right">0.1224684</td>
<td align="right">0.6061718</td>
<td align="right">1.1365265</td>
<td align="right">0.3678480</td>
<td align="right">0.00000</td>
<td align="right">0.3346527</td>
<td align="right">0.0941865</td>
<td align="right">0.0307471</td>
<td align="right">1.1535018</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">skew</td>
<td align="right">-0.9131660</td>
<td align="right">0.2921187</td>
<td align="right">-0.2621938</td>
<td align="right">0.2921187</td>
<td align="right">0.2921187</td>
<td align="right">0.2921187</td>
<td align="right">0.2921187</td>
<td align="right">NaN</td>
<td align="right">0.2921187</td>
<td align="right">-0.9580937</td>
<td align="right">-0.5895834</td>
<td align="right">0.7854057</td>
<td align="right">NaN</td>
<td align="right">0.4451142</td>
<td align="right">0.2921187</td>
<td align="right">0.2921187</td>
<td align="right">0.5167641</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">kurtosis</td>
<td align="right">-1.1430792</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2413934</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">NaN</td>
<td align="right">-2.2533333</td>
<td align="right">-1.0559053</td>
<td align="right">-1.2933074</td>
<td align="right">-1.1605394</td>
<td align="right">NaN</td>
<td align="right">-1.8437283</td>
<td align="right">-2.2533333</td>
<td align="right">-2.2533333</td>
<td align="right">-1.8277532</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">se</td>
<td align="right">0.1053033</td>
<td align="right">0.0146258</td>
<td align="right">0.1396560</td>
<td align="right">0.0302733</td>
<td align="right">0.0204192</td>
<td align="right">0.0205199</td>
<td align="right">0.0558343</td>
<td align="right">0.00000</td>
<td align="right">0.0299985</td>
<td align="right">0.1131610</td>
<td align="right">0.1860157</td>
<td align="right">0.0616354</td>
<td align="right">0.00000</td>
<td align="right">0.0621351</td>
<td align="right">0.0230709</td>
<td align="right">0.0075315</td>
<td align="right">0.2354856</td>
</tr>
<tr class="odd">
<td align="left">Fold Change</td>
<td align="left">mean</td>
<td align="right">-0.3415628</td>
<td align="right">0.0000000</td>
<td align="right">-0.2020687</td>
<td align="right">0.3430288</td>
<td align="right">0.0000000</td>
<td align="right">-1.0964727</td>
<td align="right">0.0084981</td>
<td align="right">0.00000</td>
<td align="right">0.2973479</td>
<td align="right">0.4549624</td>
<td align="right">-0.0664000</td>
<td align="right">0.3024055</td>
<td align="right">0.00000</td>
<td align="right">0.0669305</td>
<td align="right">0.0243987</td>
<td align="right">-0.5823461</td>
<td align="right">-0.9278932</td>
</tr>
<tr class="even">
<td align="left">Fold Change</td>
<td align="left">sd</td>
<td align="right">0.1803398</td>
<td align="right">0.0000000</td>
<td align="right">0.2622523</td>
<td align="right">-0.0388486</td>
<td align="right">0.0000000</td>
<td align="right">-0.1329403</td>
<td align="right">-0.1213181</td>
<td align="right">0.00000</td>
<td align="right">-0.0848902</td>
<td align="right">0.2361947</td>
<td align="right">0.3625543</td>
<td align="right">0.0723966</td>
<td align="right">0.00000</td>
<td align="right">0.0526250</td>
<td align="right">-0.1195028</td>
<td align="right">-0.0410451</td>
<td align="right">0.3219940</td>
</tr>
<tr class="odd">
<td align="left">Fold Change</td>
<td align="left">median</td>
<td align="right">-0.2561427</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.2557238</td>
<td align="right">0.0000000</td>
<td align="right">-1.2605762</td>
<td align="right">0.0000000</td>
<td align="right">0.00000</td>
<td align="right">0.2486555</td>
<td align="right">0.5004870</td>
<td align="right">0.0000000</td>
<td align="right">0.2561427</td>
<td align="right">0.00000</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">-0.6369189</td>
<td align="right">-1.1535018</td>
</tr>
<tr class="even">
<td align="left">Fold Change</td>
<td align="left">trimmed</td>
<td align="right">-0.3415628</td>
<td align="right">0.0000000</td>
<td align="right">-0.2020687</td>
<td align="right">0.3430288</td>
<td align="right">0.0000000</td>
<td align="right">-1.0964727</td>
<td align="right">0.0084981</td>
<td align="right">0.00000</td>
<td align="right">0.2973479</td>
<td align="right">0.4549624</td>
<td align="right">-0.0664000</td>
<td align="right">0.3024055</td>
<td align="right">0.00000</td>
<td align="right">0.0669305</td>
<td align="right">0.0243987</td>
<td align="right">-0.5823461</td>
<td align="right">-0.9278932</td>
</tr>
<tr class="odd">
<td align="left">Fold Change</td>
<td align="left">mad</td>
<td align="right">0.0458926</td>
<td align="right">0.0000000</td>
<td align="right">0.1354179</td>
<td align="right">-0.1396408</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">-0.3379478</td>
<td align="right">0.00000</td>
<td align="right">-0.1832349</td>
<td align="right">0.1566883</td>
<td align="right">0.1445171</td>
<td align="right">-0.0458926</td>
<td align="right">0.00000</td>
<td align="right">0.0000000</td>
<td align="right">-0.1480813</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
</tr>
<tr class="even">
<td align="left">Fold Change</td>
<td align="left">min</td>
<td align="right">-0.6881783</td>
<td align="right">0.0000000</td>
<td align="right">-0.5051718</td>
<td align="right">0.4393753</td>
<td align="right">0.0000000</td>
<td align="right">-0.9340895</td>
<td align="right">0.2293132</td>
<td align="right">0.00000</td>
<td align="right">0.3722458</td>
<td align="right">0.0307471</td>
<td align="right">-0.6855255</td>
<td align="right">0.2821618</td>
<td align="right">0.00000</td>
<td align="right">0.0000000</td>
<td align="right">0.1836515</td>
<td align="right">-0.5312341</td>
<td align="right">-1.1535018</td>
</tr>
<tr class="odd">
<td align="left">Fold Change</td>
<td align="left">max</td>
<td align="right">-0.2251885</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">0.2851276</td>
<td align="right">0.0000000</td>
<td align="right">-1.1768042</td>
<td align="right">-0.1868225</td>
<td align="right">0.00000</td>
<td align="right">0.1224684</td>
<td align="right">0.6061718</td>
<td align="right">0.3535256</td>
<td align="right">0.4923916</td>
<td align="right">0.00000</td>
<td align="right">0.1245435</td>
<td align="right">-0.1615374</td>
<td align="right">-0.6061718</td>
<td align="right">-0.4572559</td>
</tr>
<tr class="even">
<td align="left">Fold Change</td>
<td align="left">range</td>
<td align="right">0.4629898</td>
<td align="right">0.0000000</td>
<td align="right">0.5051718</td>
<td align="right">-0.1542477</td>
<td align="right">0.0000000</td>
<td align="right">-0.2427147</td>
<td align="right">-0.4161357</td>
<td align="right">0.00000</td>
<td align="right">-0.2497773</td>
<td align="right">0.5754247</td>
<td align="right">1.0390511</td>
<td align="right">0.2102298</td>
<td align="right">0.00000</td>
<td align="right">0.1245435</td>
<td align="right">-0.3451889</td>
<td align="right">-0.0749377</td>
<td align="right">0.6962459</td>
</tr>
<tr class="odd">
<td align="left">Fold Change</td>
<td align="left">skew</td>
<td align="right">-0.6210473</td>
<td align="right">0.0000000</td>
<td align="right">-0.5543125</td>
<td align="right">0.4451286</td>
<td align="right">0.0000000</td>
<td align="right">0.5842374</td>
<td align="right">0.1823297</td>
<td align="right">NaN</td>
<td align="right">-0.3641815</td>
<td align="right">-0.6659750</td>
<td align="right">-0.2974647</td>
<td align="right">1.7115331</td>
<td align="right">NaN</td>
<td align="right">-0.3882984</td>
<td align="right">0.0647746</td>
<td align="right">0.5842374</td>
<td align="right">-0.0850910</td>
</tr>
<tr class="even">
<td align="left">Fold Change</td>
<td align="left">kurtosis</td>
<td align="right">1.1102542</td>
<td align="right">0.0000000</td>
<td align="right">0.0119400</td>
<td align="right">-0.4557933</td>
<td align="right">0.0000000</td>
<td align="right">0.0000000</td>
<td align="right">-0.4611003</td>
<td align="right">NaN</td>
<td align="right">-0.8567128</td>
<td align="right">1.1974280</td>
<td align="right">0.9600259</td>
<td align="right">-0.0674730</td>
<td align="right">NaN</td>
<td align="right">-0.6442716</td>
<td align="right">-0.4591298</td>
<td align="right">0.0000000</td>
<td align="right">-0.1543832</td>
</tr>
<tr class="odd">
<td align="left">Fold Change</td>
<td align="left">se</td>
<td align="right">0.0806504</td>
<td align="right">0.0000000</td>
<td align="right">0.1172828</td>
<td align="right">-0.0173736</td>
<td align="right">0.0000000</td>
<td align="right">-0.0594527</td>
<td align="right">-0.0542551</td>
<td align="right">0.00000</td>
<td align="right">-0.0379641</td>
<td align="right">0.1056295</td>
<td align="right">0.1621392</td>
<td align="right">0.0323767</td>
<td align="right">0.00000</td>
<td align="right">0.0235346</td>
<td align="right">-0.0534433</td>
<td align="right">-0.0183559</td>
<td align="right">0.1440001</td>
</tr>
</tbody>
</table>
</div>
<p>The column “Stat” in the generated result includes the following
statistics:</p>
<ul>
<li>n: number.</li>
<li>mean: mean.</li>
<li>sd: standard deviation.</li>
<li>median: median.</li>
<li>trimmed: trimmed mean with a trim of 0.1.</li>
<li>mad: median absolute deviation (from the median).</li>
<li>min: minimum.</li>
<li>max: maximum.</li>
<li>range: the difference between the maximum and minimum value.</li>
<li>skew: skewness.</li>
<li>kurtosis: kurtosis.</li>
<li>se: standard error.</li>
</ul>
</div>
<div class="section level2">
<h2 id="analysis">Analysis<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<p>The function <code><a href="../reference/analyze.html">analyze()</a></code> calculates the results that can
be used in subsequent visualizations. If more than two conditions exist
in the data, precisely two conditions for comparison must be specified
via the argument <code>conditions</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cond</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"50fmol"</span>, <span class="st">"100fmol"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="students-t-test">Student’s t-test<a class="anchor" aria-label="anchor" href="#students-t-test"></a>
</h3>
<p>The Student’s t-test is used to compare the means between two
conditions for each protein, reporting both the difference in means
between the conditions (calculated as Condition 1 - Condition 2) and the
P-value of the test.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anlys_t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze.html">analyze</a></span><span class="op">(</span><span class="va">dataImput</span>, conditions <span class="op">=</span> <span class="va">cond</span>, testType <span class="op">=</span> <span class="st">"t-test"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Data are essentially constant.</span></span>
<span><span class="co">#&gt; Data are essentially constant.</span></span></code></pre></div>
<p>Oops! The warning message shows “Data are essentially constant,”
which means that the data contain proteins with the same value in all
samples. In this case, the P-value of t-test returns NA.</p>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Difference</td>
<td align="right">-0.3415628</td>
<td align="right">0</td>
<td align="right">-0.2020687</td>
<td align="right">0.3430288</td>
<td align="right">0</td>
<td align="right">-1.0964727</td>
<td align="right">0.0084981</td>
<td align="right">0</td>
<td align="right">0.2973479</td>
<td align="right">0.4549624</td>
<td align="right">-0.0664000</td>
<td align="right">0.3024055</td>
<td align="right">0</td>
<td align="right">0.0669305</td>
<td align="right">0.0243987</td>
<td align="right">-0.5823461</td>
<td align="right">-0.9278932</td>
</tr>
<tr class="even">
<td align="left">P-value</td>
<td align="right">0.0296551</td>
<td align="right">1</td>
<td align="right">0.2229593</td>
<td align="right">0.0005694</td>
<td align="right">1</td>
<td align="right">0.0000852</td>
<td align="right">0.9473757</td>
<td align="right">NA</td>
<td align="right">0.0084762</td>
<td align="right">0.0156987</td>
<td align="right">0.7406309</td>
<td align="right">0.0049582</td>
<td align="right">NA</td>
<td align="right">0.3919954</td>
<td align="right">0.7731339</td>
<td align="right">0.0000073</td>
<td align="right">0.0135214</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="moderated-t-test">Moderated t-test<a class="anchor" aria-label="anchor" href="#moderated-t-test"></a>
</h3>
<p>The main distinction between the Student’s and moderated t-tests
<span class="citation">(Smyth 2004)</span> lies in how variance is
computed. While the Student’s t-test calculates variance based on the
data available for each protein individually, the moderated t-test
utilizes information from all the chosen proteins to calculate
variance.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anlys_mod.t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze.html">analyze</a></span><span class="op">(</span><span class="va">dataImput</span>, conditions <span class="op">=</span> <span class="va">cond</span>, testType <span class="op">=</span> <span class="st">"mod.t-test"</span><span class="op">)</span></span></code></pre></div>
<p>In the moderated t-test, a warning message might occur stating, “Zero
sample variances detected, have been offset away from zero.” This
warning corresponds to examples of proteins that exhibited identical
quant values, either pre- or post-imputation, and therefore no variance
is present across conditions for those proteins. This does not impede
downstream analysis; it merely serves to alert users to its
occurrence.</p>
<!-- This just means that for at least one protein the log ratio is identical for all samples. -->
<!-- Since this will give a zero variance (which will end up in the denominator of your -->
<!-- statistic and could possibly result in an infinite value for your test statistic) it has -->
<!-- been offset to a small value to prevent that possibility. -->
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Difference</td>
<td align="right">-0.3415628</td>
<td align="right">0</td>
<td align="right">-0.2020687</td>
<td align="right">0.3430288</td>
<td align="right">0</td>
<td align="right">-1.0964727</td>
<td align="right">0.0084981</td>
<td align="right">0</td>
<td align="right">0.2973479</td>
<td align="right">0.4549624</td>
<td align="right">-0.0664000</td>
<td align="right">0.3024055</td>
<td align="right">0</td>
<td align="right">0.0669305</td>
<td align="right">0.0243987</td>
<td align="right">-0.5823461</td>
<td align="right">-0.9278932</td>
</tr>
<tr class="even">
<td align="left">P-value</td>
<td align="right">0.0106414</td>
<td align="right">1</td>
<td align="right">0.1765624</td>
<td align="right">0.0001868</td>
<td align="right">1</td>
<td align="right">0.0000004</td>
<td align="right">0.9450320</td>
<td align="right">1</td>
<td align="right">0.0028978</td>
<td align="right">0.0028530</td>
<td align="right">0.7238526</td>
<td align="right">0.0015565</td>
<td align="right">1</td>
<td align="right">0.3715159</td>
<td align="right">0.7605007</td>
<td align="right">0.0000000</td>
<td align="right">0.0047454</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="ma">MA<a class="anchor" aria-label="anchor" href="#ma"></a>
</h3>
<p>The result of <code>testType = "MA"</code> is to generate the data
for plotting an MA plot, which represents the protein-wise averages
within each condition.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anlys_MA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/analyze.html">analyze</a></span><span class="op">(</span><span class="va">dataImput</span>, conditions <span class="op">=</span> <span class="va">cond</span>, testType <span class="op">=</span> <span class="st">"MA"</span><span class="op">)</span></span></code></pre></div>
<div style="overflow-x: auto;">
<table style="width:100%;" class="table">
<colgroup>
<col width="3%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="4%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">RAB3D_HUMAN</th>
<th align="right">ADH1_YEAST</th>
<th align="right">LYSC_CHICK</th>
<th align="right">BGAL_HUMAN</th>
<th align="right">SYTC_HUMAN</th>
<th align="right">CYC_BOVIN</th>
<th align="right">PA1B2_HUMAN</th>
<th align="right">TEBP_HUMAN</th>
<th align="right">UAP1_HUMAN</th>
<th align="right">B3GLT_HUMAN</th>
<th align="right">NFXL1_HUMAN</th>
<th align="right">VPS36_HUMAN</th>
<th align="right">T126B_HUMAN</th>
<th align="right">ORC3_HUMAN</th>
<th align="right">BAG5_HUMAN</th>
<th align="right">ANGL3_HUMAN</th>
<th align="right">ZC11B_HUMAN</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="right">16.77585</td>
<td align="right">20.02791</td>
<td align="right">19.52968</td>
<td align="right">17.87318</td>
<td align="right">21.25237</td>
<td align="right">18.85738</td>
<td align="right">17.39531</td>
<td align="right">22.12204</td>
<td align="right">18.29065</td>
<td align="right">14.62582</td>
<td align="right">15.9883</td>
<td align="right">16.54702</td>
<td align="right">13.59442</td>
<td align="right">16.44383</td>
<td align="right">17.92432</td>
<td align="right">15.20201</td>
<td align="right">17.22063</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="right">16.43429</td>
<td align="right">20.02791</td>
<td align="right">19.32761</td>
<td align="right">18.21620</td>
<td align="right">21.25237</td>
<td align="right">17.76090</td>
<td align="right">17.40381</td>
<td align="right">22.12204</td>
<td align="right">18.58800</td>
<td align="right">15.08078</td>
<td align="right">15.9219</td>
<td align="right">16.84942</td>
<td align="right">13.59442</td>
<td align="right">16.51076</td>
<td align="right">17.94872</td>
<td align="right">14.61967</td>
<td align="right">16.29274</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<p>This section provides a variety of options for getting a global view
of your data, making comparisons, and highlighting trends. Keep in mind
that data visualization is most effective when illustrating a point or
answering a question you have about your data, and not as a means to
find a point/question.</p>
<div class="section level3">
<h3 id="heatmap">heatmap<a class="anchor" aria-label="anchor" href="#heatmap"></a>
</h3>
<p>The package offers two options for plotting the heatmap.</p>
<ul>
<li>Option 1 utilizes the source package <code>pheatmap</code>, capable
of plotting the dendrogram simultaneously. It is the default choice for
heatmaps in this package.</li>
</ul>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataImput</span>, graphType <span class="op">=</span> <span class="st">"heatmap"</span>,</span>
<span>          pkg <span class="op">=</span> <span class="st">"pheatmap"</span>,</span>
<span>          cluster_cols <span class="op">=</span> <span class="cn">TRUE</span>, cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>          show_colnames <span class="op">=</span> <span class="cn">TRUE</span>, show_rownames <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-31-1.png" width="700" style="display: block; margin: auto;"></p>
<p>When protein names are excessively long, it is recommended to set
<code>show_rownames = FALSE</code> to view the full heatmap.</p>
<ul>
<li>Option 2 use the source package <code>ggplot2</code> to generate a
ggplot object but does not include the dendrogram.</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataImput</span>, graphType <span class="op">=</span> <span class="st">"heatmap"</span>, pkg <span class="op">=</span> <span class="st">"ggplot2"</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-32-1.png" width="700" style="display: block; margin: auto;"></p>
<p>In a heatmap, similar colors within a row indicate relatively
consistent values, suggesting similar protein expression levels across
different samples.</p>
</div>
<div class="section level3">
<h3 id="ma-1">MA<a class="anchor" aria-label="anchor" href="#ma-1"></a>
</h3>
<!-- An MA plot visualizes the differences between measurements taken in two samples, -->
<!-- by transforming the data onto M (log ratio or fold change) and A (mean average) scales. -->
<!-- The MA plot puts the variable M on the y-axis and A on the x-axis and gives a quick -->
<!-- overview of the distribution of the data. Most proteins are expected to show little -->
<!-- variation, and so the majority of the points will be concentrated around the M = 0 line -->
<!-- (no difference between group means). Typically, points falling above $abs(1)$ are -->
<!-- highlighted. -->
<p>An MA plot, short for “M vs. A plot,” which uses two axes:</p>
<ul>
<li>M axis (vertical): Represents the logarithm (usually base 2) of the
fold change, or the ratio of the expression levels, between two
conditions. It is calculated as: <span class="math display">\[M =
log_2(X/Y) = log_2 X - log_2 Y\]</span>
</li>
<li>A axis (horizontal): Represents the average intensity of the two
conditions, calculated as: <span class="math display">\[A =
\frac{1}{2}log_2(XY) =
\frac{1}{2}\left[log_2(X)+log_2(Y)\right]\]</span>
</li>
</ul>
<!-- So, the M value represents the log ratio of protein expression in a treated condition --><!-- compared to a control condition, where higher M values indicate greater differences in --><!-- expression between the conditions. Meanwhile, the A value helps to account for overall --><!-- differences in expression levels and reduce the influence of technical biases. --><p>Most proteins are expected to exhibit little variation, leading to
the majority of points concentrating around the line M = 0 (indicating
no difference between group means).</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">anlys_MA</span>, graphType <span class="op">=</span> <span class="st">"MA"</span>, M.thres <span class="op">=</span> <span class="fl">1</span>, transformLabel <span class="op">=</span> <span class="st">"Log2"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 16 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-33-1.png" width="700" style="display: block; margin: auto;"></p>
<p>where <code>M.thres = 1</code> means the M thresholds are set to −1
and 1. The scatters are split into three parts: significant up (M &gt;
1), no significant (-1 <span class="math inline">\(\leq\)</span> M <span class="math inline">\(\leq\)</span> 1), and significant down (M &lt;
-1). And <code>transformLabel = "Log2"</code> is used to label the
title. Additionally, the warning message “Removed 16 rows containing
missing values” indicates that there are 16 proteins with no
significance.</p>
</div>
<div class="section level3">
<h3 id="normalize">Normalize<a class="anchor" aria-label="anchor" href="#normalize"></a>
</h3>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataNorm</span>, graphType <span class="op">=</span> <span class="st">"normalize"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 16 rows containing non-finite outside the scale range</span></span>
<span><span class="co">#&gt; (`stat_boxplot()`).</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-34-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="pca">PCA<a class="anchor" aria-label="anchor" href="#pca"></a>
</h3>
<p>Principal component analysis (PCA) is a powerful technique used in
data analysis to simplify and reduce the dimensionality of large
datasets. It transforms original variables into uncorrelated components
that capture the maximum variance. By selecting a subset of these
components, PCA projects the data points onto these key directions,
enabling visualization and analysis in a lower-dimensional space. This
aids in identifying patterns and relationships within complex
datasets.</p>
<p>In the visualization for <code>graphType = "PCA_*"</code>, the
arguments <code>center</code> and <code>scale</code> are used to center
the data to zero mean and scale to unit variance, with default setting
at <code>TRUE</code>. The most common error message is “Cannot rescale a
constant/zero column to unit variance.” This clearly occurs when columns
representing proteins contain only zeros or have constant values.
Typically, there are two ways to address this error: one is to remove
these proteins, and the other is to set <code>scale = FALSE</code>.</p>
<p><strong>NOTE:</strong> Data scaling is done to ensure that the scale
differences between different features do not affect the results of PCA.
If not scaled, features with larger scales will dominate the computation
of principal components (PCs).</p>
<p>In the case of <code>dataImput</code>, two proteins, namely
“TEBP_HUMAN” and “T126B_HUMAN,” have constant values, leading to the
error message. We choose to remove these two proteins in PCA.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataPCA</span> <span class="op">&lt;-</span> <span class="va">dataImput</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">dataImput</span><span class="op">)</span> <span class="op">!=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TEBP_HUMAN"</span>, <span class="st">"T126B_HUMAN"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<div class="section level4">
<h4 id="pca_scree">PCA_scree<a class="anchor" aria-label="anchor" href="#pca_scree"></a>
</h4>
<p>One way to help identify how many PCs to retain, is to explore a
scree plot. The scree plot shows the eigenvalues of each PC, which
represent the proportion of variance explained by that component.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataPCA</span>, graphType <span class="op">=</span> <span class="st">"PCA_scree"</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          addlabels <span class="op">=</span> <span class="cn">TRUE</span>, choice <span class="op">=</span> <span class="st">"variance"</span>, ncp <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataPCA</span>, graphType <span class="op">=</span> <span class="st">"PCA_scree"</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          addlabels <span class="op">=</span> <span class="cn">TRUE</span>, choice <span class="op">=</span> <span class="st">"eigenvalue"</span>, ncp <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>where <code>choice</code> specifies the data to be plotted, either
<code>"variance"</code> or <code>"eigenvalue"</code>,
<code>addlabels = TRUE</code> adds information labels at the top of
bars/points, and <code>ncp = 10</code> sets the number of dimension to
be displayed.</p>
<p><img src="usage_template_files/figure-html/unnamed-chunk-37-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="pca_ind">PCA_ind<a class="anchor" aria-label="anchor" href="#pca_ind"></a>
</h4>
<p>The primary PCA plot of individual data visually represents the
distribution of individual observations in a reduced-dimensional space,
typically defined by the PCs. The x and y axes of the PCA plot represent
the PCs. Each axis corresponds to a linear combination of the original
variables. Individual data points on the PCA plot represent observations
(e.g., samples) from the original dataset. Points that are close to the
origin (0, 0), are close to the “average” across all protein abundances.
If sufficient samples are present, the plot will also produce a 95%
confidence ellipse, as well as a centroid (mean for each group
provided), for each groups (condition) provided.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataPCA</span>, graphType <span class="op">=</span> <span class="st">"PCA_ind"</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          addlabels <span class="op">=</span> <span class="cn">TRUE</span>, addEllipses <span class="op">=</span> <span class="cn">TRUE</span>, ellipse.level <span class="op">=</span> <span class="fl">0.95</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-38-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="pca_var">PCA_var<a class="anchor" aria-label="anchor" href="#pca_var"></a>
</h4>
<p>This plot will be more useful if your analyses are based on a
relatively small number of proteins. It represents the association, or
loading of each protein on the first two PCs. Longer arrows represents
stronger associations. It is important to note that proteins that are
weakly associated with PC1 or PC2 may still be highly correlated with
other PCs not being plotted. Consult the scree plot (and other available
methods) to determine the appropriate number of PCs to investigate.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataPCA</span>, graphType <span class="op">=</span> <span class="st">"PCA_var"</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          addlabels <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-39-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="pca_biplot">PCA_biplot<a class="anchor" aria-label="anchor" href="#pca_biplot"></a>
</h4>
<p>The PCA biplot includes individual and variable plots. Again, with a
large number of proteins, this plot can be unwieldy.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataPCA</span>, graphType <span class="op">=</span> <span class="st">"PCA_biplot"</span>, center <span class="op">=</span> <span class="cn">TRUE</span>, scale <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          addEllipses <span class="op">=</span> <span class="cn">TRUE</span>, ellipse.level <span class="op">=</span> <span class="fl">0.95</span>, label <span class="op">=</span> <span class="st">"all"</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-40-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level3">
<h3 id="t-test">t-test<a class="anchor" aria-label="anchor" href="#t-test"></a>
</h3>
<p>The function <code><a href="../reference/visualize.html">visualize()</a></code> can be applied to any t-test
output. It generates two useful plots: a histogram of fold changes
across the analyzed proteins and a histogram of P-values. The majority
of proteins are expected to show very small change between conditions,
so the fold change histogram will have a peak at around zero. For the
P-values, most P-values are expected to be non-significant (above 0.05).
Depending on the strength of the treatment effect, there may be a peak
of p-values near 0.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">anlys_mod.t</span>, graphType <span class="op">=</span> <span class="st">"t-test"</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-41-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="upset">Upset<a class="anchor" aria-label="anchor" href="#upset"></a>
</h3>
<p>The upset plot is a visual representation that helps display the
overlap and intersection of sets or categories in a dataset. It is
particularly useful for illustrating the presence or absence of elements
in combinations of sets.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dataSort</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sortcondition.html">sortcondition</a></span><span class="op">(</span><span class="va">dataSet</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataSort</span>, graphType <span class="op">=</span> <span class="st">"Upset"</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-42-1.png" width="700" style="display: block; margin: auto;"></p>
<p>This plot reveals that 18 proteins are in common between 100fmol and
50fmol, while only 1 protein is unique to 100fmol.</p>
</div>
<div class="section level3">
<h3 id="venn">Venn<a class="anchor" aria-label="anchor" href="#venn"></a>
</h3>
<p>The Venn plot is another graphical representation of the
relationships between sets. Each circle represents a set, and the
overlapping regions show the elements that are shared between sets.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">dataSort</span>, graphType <span class="op">=</span> <span class="st">"Venn"</span>,</span>
<span>          show_percentage <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>          fill_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blue"</span>, <span class="st">"yellow"</span>, <span class="st">"green"</span>, <span class="st">"red"</span><span class="op">)</span>,</span>
<span>          show_universal <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-43-1.png" width="700" style="display: block; margin: auto;"></p>
<p>In the example above, 100fmol and 50fmol groups share 18 proteins.
Notably, one protein is exclusively found in the 100fmol group, while
the 50fmol group lacks any unique proteins.</p>
</div>
<div class="section level3">
<h3 id="volcano">Volcano<a class="anchor" aria-label="anchor" href="#volcano"></a>
</h3>
<p>A volcano plot is a graphical representation commonly used in
proteomics and genomics to visualize differential expression analysis
results. It is particularly useful for identifying significant changes
in extensive data. It displays two important pieces of information about
differences between conditions in a dataset:</p>
<ul>
<li><p>Statistical significance (vertical): Represents the negative
log10 of the P-value.</p></li>
<li><p>Fold change (horizontal): Represents the fold change.</p></li>
</ul>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/visualize.html">visualize</a></span><span class="op">(</span><span class="va">anlys_mod.t</span>, graphType <span class="op">=</span> <span class="st">"volcano"</span>,</span>
<span>          P.thres <span class="op">=</span> <span class="fl">0.05</span>, logF.thres <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Removed 15 rows containing missing values or values outside the scale range</span></span>
<span><span class="co">#&gt; (`geom_text_repel()`).</span></span></code></pre></div>
<p><img src="usage_template_files/figure-html/unnamed-chunk-44-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level2">
<h2 id="other-useful-function">Other useful function<a class="anchor" aria-label="anchor" href="#other-useful-function"></a>
</h2>
<p>The function <code><a href="../reference/pullProteinPath.html">pullProteinPath()</a></code> allows you to see the
values associated with a specific protein at each step of processing.
This can be useful for questions such as, “Were all of the values for my
favorite protein actually measured, or were some imputed?” or “Why
didn’t my favorite protein make it to the final list? Where was it
filtered out?”. It can also be used to check whether a given protein’s
fold-change might have been a processing artifact.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ZC11B</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pullProteinPath.html">pullProteinPath</a></span><span class="op">(</span>proteinName <span class="op">=</span> <span class="st">"ZC11B_HUMAN"</span>,</span>
<span>                         dataSetList <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>Initial <span class="op">=</span> <span class="va">dataSet</span>,</span>
<span>                                            Transformed <span class="op">=</span> <span class="va">dataTran</span>,</span>
<span>                                            Normalized <span class="op">=</span> <span class="va">dataNorm</span>,</span>
<span>                                            Imputed <span class="op">=</span> <span class="va">dataImput</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div style="overflow-x: auto;">
<table class="table">
<colgroup>
<col width="9%">
<col width="49%">
<col width="9%">
<col width="7%">
<col width="9%">
<col width="8%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left">R.Condition</th>
<th align="left">R.FileName</th>
<th align="right">R.Replicate</th>
<th align="right">Initial</th>
<th align="right">Transformed</th>
<th align="right">Normalized</th>
<th align="right">Imputed</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">17.08332</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">17.08332</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">172537.11</td>
<td align="right">17.39655</td>
<td align="right">17.08332</td>
<td align="right">17.08332</td>
</tr>
<tr class="even">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">172585.23</td>
<td align="right">17.39695</td>
<td align="right">17.54057</td>
<td align="right">17.54057</td>
</tr>
<tr class="odd">
<td align="left">100fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_100fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">136210.98</td>
<td align="right">17.05548</td>
<td align="right">17.31263</td>
<td align="right">17.31263</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_1</td>
<td align="right">1</td>
<td align="right">60972.31</td>
<td align="right">15.89587</td>
<td align="right">15.92981</td>
<td align="right">15.92981</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_2</td>
<td align="right">2</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">15.92981</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_3</td>
<td align="right">3</td>
<td align="right">112139.66</td>
<td align="right">16.77494</td>
<td align="right">16.59092</td>
<td align="right">16.59092</td>
</tr>
<tr class="odd">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_4</td>
<td align="right">4</td>
<td align="right">109480.65</td>
<td align="right">16.74032</td>
<td align="right">17.08332</td>
<td align="right">17.08332</td>
</tr>
<tr class="even">
<td align="left">50fmol</td>
<td align="left">20230414_jcl_msDIAlogue_HeLa-6mix_50fmol_60min_DIA_Wat25cmBEH_5</td>
<td align="right">5</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">NA</td>
<td align="right">15.92981</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="reference">Reference<a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-bolstad2003comparison" class="csl-entry">
Bolstad, B. M., R. A. Irizarry, M. Astrand, and T. P. Speed. 2003.
<span>“A Comparison of Normalization Methods for High Density
Oligonucleotide Array Data Based on Variance and Bias.”</span>
<em>Bioinformatics</em> 19 (2): 185–93. <a href="https://doi.org/10.1093/bioinformatics/19.2.185" class="external-link">https://doi.org/10.1093/bioinformatics/19.2.185</a>.
</div>
<div id="ref-breiman1984classification" class="csl-entry">
Breiman, L., J. Friedman, R. A. Olshen, and C. J. Stone. 1984.
<em>Classification and Regression Trees</em>. New York, NY, USA:
Routledge.
</div>
<div id="ref-doove2014recursive" class="csl-entry">
Doove, Lisa L., Stef van Buuren, and Elise Dusseldorp. 2014.
<span>“Recursive Partitioning for Missing Data Imputation in the
Presence of Interaction Effects.”</span> <em>Computational Statistics
&amp; Data Analysis</em> 72: 92–104. <a href="https://doi.org/10.1016/j.csda.2013.10.025" class="external-link">https://doi.org/10.1016/j.csda.2013.10.025</a>.
</div>
<div id="ref-hastie2015matrix" class="csl-entry">
Hastie, Trevor, Rahul Mazumder, Jason D. Lee, and Reza Zadeh. 2015.
<span>“Matrix Completion and Low-Rank <span>SVD</span> via Fast
Alternating Least Squares.”</span> <em>Journal of Machine Learning
Research</em> 16 (104): 3367—3402. <a href="http://jmlr.org/papers/v16/hastie15a.html" class="external-link">http://jmlr.org/papers/v16/hastie15a.html</a>.
</div>
<div id="ref-kim2004reuse" class="csl-entry">
Kim, Ki-Yeol, Byoung-Jin Kim, and Gwan-Su Yi. 2004. <span>“Reuse of
Imputed Data in Microarray Analysis Increases Imputation
Efficiency.”</span> <em>BMC Bioinformatics</em> 5: 160. <a href="https://doi.org/10.1186/1471-2105-5-160" class="external-link">https://doi.org/10.1186/1471-2105-5-160</a>.
</div>
<div id="ref-oba2003bayesian" class="csl-entry">
Oba, Shigeyuki, Masa-aki Sato, Ichiro Takemasa, Morito Monden, Ken-ichi
Matsubara, and Shin Ishii. 2003. <span>“A <span>B</span>ayesian Missing
Value Estimation Method for Gene Expression Profile Data.”</span>
<em>Bioinformatics</em> 19 (16): 2088–96. <a href="https://doi.org/10.1093/bioinformatics/btg287" class="external-link">https://doi.org/10.1093/bioinformatics/btg287</a>.
</div>
<div id="ref-rubin1987multiple" class="csl-entry">
Rubin, Donald B. 1987. <em>Multiple Imputation for Nonresponse in
Surveys</em>. New York, NY, USA: John Wiley &amp; Sons.
</div>
<div id="ref-schafer1997analysis" class="csl-entry">
Schafer, Joseph L. 1997. <em>Analysis of Incomplete Multivariate
Data</em>. New York, NY, USA: Chapman &amp; Hall/CRC.
</div>
<div id="ref-shah2017distribution" class="csl-entry">
Shah, Jasmit S., Shesh N. Rai, Andrew P. DeFilippis, Bradford G. Hill,
Aruni Bhatnagar, and Guy N. Brock. 2017. <span>“Distribution Based
Nearest Neighbor Imputation for Truncated High Dimensional Data with
Applications to Pre-Clinical and Clinical Metabolomics Studies.”</span>
<em>BMC Bioinformatics</em> 18: 114. <a href="https://doi.org/10.1186/s12859-017-1547-6" class="external-link">https://doi.org/10.1186/s12859-017-1547-6</a>.
</div>
<div id="ref-smyth2004linear" class="csl-entry">
Smyth, Gordon K. 2004. <span>“Linear Models and Empirical Bayes Methods
for Assessing Differential Expression in Microarray Experiments.”</span>
<em>Statistical Applications in Genetics and Molecular Biology</em> 3
(1). <a href="https://doi.org/10.2202/1544-6115.1027" class="external-link">https://doi.org/10.2202/1544-6115.1027</a>.
</div>
<div id="ref-stacklies2007pcamethods" class="csl-entry">
Stacklies, Wolfram, Henning Redestig, Matthias Scholz, Dirk Walther, and
Joachim Selbig. 2007. <span>“<span class="nocase">pcaMethods</span>–a
Bioconductor Package Providing <span>PCA</span> Methods for Incomplete
Data.”</span> <em>Bioinformatics</em> 23 (9): 1164–67. <a href="https://doi.org/10.1093/bioinformatics/btm069" class="external-link">https://doi.org/10.1093/bioinformatics/btm069</a>.
</div>
<div id="ref-troyanskaya2001missing" class="csl-entry">
Troyanskaya, Olga, Michael Cantor, Gavin Sherlock, Pat Brown, Trevor
Hastie, Robert Tibshirani, David Botstein, and Russ B. Altman. 2001.
<span>“Missing Value Estimation Methods for <span>DNA</span>
Microarrays.”</span> <em>Bioinformatics</em> 17 (6): 520–25. <a href="https://doi.org/10.1093/bioinformatics/17.6.520" class="external-link">https://doi.org/10.1093/bioinformatics/17.6.520</a>.
</div>
<div id="ref-van2018flexible" class="csl-entry">
van Buuren, Stef. 2018. <em>Flexible Imputation of Missing Data</em>.
New York, NY, USA: Chapman &amp; Hall/CRC.
</div>
<div id="ref-van2011mice" class="csl-entry">
van Buuren, Stef, and Karin Groothuis-Oudshoorn. 2011. <span>“Mice:
<span>M</span>ultivariate Imputation by Chained Equations in
<span>R</span>.”</span> <em>Journal of Statistical Software</em> 45 (3):
1–67. <a href="https://doi.org/10.18637/jss.v045.i03" class="external-link">https://doi.org/10.18637/jss.v045.i03</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://shiying-xiao.com" class="external-link">Shiying Xiao</a>, <a href="https://statsconsulting.uconn.edu/the-consulting-team" class="external-link">Timothy Moore</a>, Charles Watt.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
